{
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "view-in-github",
        "colab_type": "text"
      },
      "source": [
        "<a href=\"https://colab.research.google.com/github/Thiiagoo777/Mielo1/blob/main/PIPELINE_FILTRAGEM_DE_RISCO_MF.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "-sX7Ic9zOtZe"
      },
      "source": [
        "## **FILTRAGEM DE VARIANTES SOMÁTICAS - PAINEL MIELOFIBROSE PROGNÓSTICO RUIM**\n",
        "\n",
        "**OBJETIVO**: Este *pipeline* se destina à filtrar variantes somáticas que ofereçam prognóstico ruim para a doença de mielofibrose.\n",
        "\n",
        "- Foram filtradas 30 amostras do projeto \"LMA Brasil\".\n",
        "\n",
        "- Os genes considerados para a filtragem foram:\n",
        "\n",
        "- CALR, ASXL1, SRSF2, U2AF1 (baseado em artigo fundador do GIPSS – sistema de prognóstico por pontuação inspirado em dados genéticos https://doi.org/10.1038/s41375-018-0107-z)"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "p-Qt8gxxmB3N"
      },
      "source": []
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "fHH1jk0rKrb0"
      },
      "source": [
        "## 1. Preparar ambiente de trabalho\n",
        "\n",
        "\n",
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "L75WGIUQAa1m",
        "outputId": "e3517f3f-a58d-4fe0-b2eb-28ca06594fd3"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "Cloning into 'lmabrasil-hg38'...\n"
          ]
        }
      ],
      "source": [
        "#Clona github do projeto lmabrasil-hg48.git\n",
        "%%bash\n",
        "rm -rf lmabrasil-hg38\n",
        "git clone https://github.com/renatopuga/lmabrasil-hg38"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "sUoh_CpV7SAH",
        "outputId": "00962988-2ada-447d-9b73-aa67f33e2edb"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Cloning into 'projetolma'...\n",
            "remote: Enumerating objects: 35, done.\u001b[K\n",
            "remote: Counting objects: 100% (35/35), done.\u001b[K\n",
            "remote: Compressing objects: 100% (34/34), done.\u001b[K\n",
            "remote: Total 35 (delta 29), reused 0 (delta 0), pack-reused 0\u001b[K\n",
            "Receiving objects: 100% (35/35), 28.22 MiB | 2.79 MiB/s, done.\n",
            "Resolving deltas: 100% (29/29), done.\n"
          ]
        }
      ],
      "source": [
        "#Clona meu github contendo amostras do projeto lmabrasil convertidos para hg38\n",
        "!git clone https://github.com/bioinfoamos01/projetolma.git"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "Rckp-nMX7cHc"
      },
      "outputs": [],
      "source": [
        "#remove arquivo README.md da pasta lma\n",
        "!rm /content/projetolma/README.md"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "P4BOezJ2AHeD"
      },
      "outputs": [],
      "source": [
        "#Copia 30 arquivos da pasta projetolma(contendo amostras pós lift-over hg38) para a pasta /vep_output\n",
        "!cp /content/projetolma/* /content/lmabrasil-hg38/vep_output"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "Zko2ES3zJHFQ"
      },
      "outputs": [],
      "source": [
        "#Cria uma lista.txt com 9 genes de risco para prognóstico ruim para mielofibrose\n",
        "!echo -e \"CALR\\nASXL1\\nSRSF2\\nEZH2\\nIDH1\\nIDH2\\nU2AF1\\n\" > /content/lmabrasil-hg38/hpo/mielofibrose.txt"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "t2S29DCVr9GV"
      },
      "outputs": [],
      "source": []
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "FuF92mtzKwzB"
      },
      "source": [
        "## 2. Instalar bcftools com plugin (+split-vep)\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "YRafP0HDQI3q",
        "outputId": "23a8b9e9-944a-48ae-b128-808eb83c09d3"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Submodule path 'htscodecs': checked out 'ffda7310c4b3292955561d6c3b1743cb82bfe26b'\n",
            "cd ../htslib && make htslib.pc.tmp\n",
            "make[1]: Entering directory '/content/htslib'\n",
            "echo '# Default htscodecs.mk generated by Makefile' > htscodecs.mk\n",
            "echo 'include $(HTSPREFIX)htscodecs_bundled.mk' >> htscodecs.mk\n",
            "./hts_probe_cc.sh 'gcc' '-g -Wall -O2 -fvisibility=hidden ' '-fvisibility=hidden' >> htscodecs.mk\n",
            "sed -e '/^static_libs=/s/@static_LIBS@/-lz -lm -lbz2 -llzma -lcurl/;s#@[^-][^@]*@##g' htslib.pc.in > htslib.pc.tmp\n",
            "make[1]: Leaving directory '/content/htslib'\n",
            "cd ../htslib && make htslib_static.mk\n",
            "make[1]: Entering directory '/content/htslib'\n",
            "sed -n '/^static_libs=/s/[^=]*=/HTSLIB_static_LIBS = /p;/^static_ldflags=/s/[^=]*=/HTSLIB_static_LDFLAGS = /p' htslib.pc.tmp > htslib_static.mk\n",
            "make[1]: Leaving directory '/content/htslib'\n",
            "echo '/* Basic config.h generated by Makefile */' > config.h\n",
            "echo '#define ENABLE_BCF_PLUGINS 1' >> config.h\n",
            "echo '#define PLUGIN_EXT \".so\"' >> config.h\n",
            "echo '#define BCFTOOLS_VERSION \"1.19-26-g8e998eb6\"' > version.h\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o main.o main.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o vcfindex.o vcfindex.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o tabix.o tabix.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o vcfstats.o vcfstats.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o vcfisec.o vcfisec.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o vcfmerge.o vcfmerge.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o vcfquery.o vcfquery.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o vcffilter.o vcffilter.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib    -c -o filter.o filter.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o vcfsom.o vcfsom.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o vcfnorm.o vcfnorm.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o vcfgtcheck.o vcfgtcheck.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o vcfview.o vcfview.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o vcfannotate.o vcfannotate.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o vcfroh.o vcfroh.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o vcfconcat.o vcfconcat.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o vcfcall.o vcfcall.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o mcall.o mcall.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o vcmp.o vcmp.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o gvcf.o gvcf.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o reheader.o reheader.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o convert.o convert.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o vcfconvert.o vcfconvert.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o tsv2vcf.o tsv2vcf.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o vcfcnv.o vcfcnv.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o vcfhead.o vcfhead.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o HMM.o HMM.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o consensus.o consensus.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o ploidy.o ploidy.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o bin.o bin.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o hclust.o hclust.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o version.o version.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o regidx.o regidx.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o smpl_ilist.o smpl_ilist.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o csq.o csq.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o vcfbuf.o vcfbuf.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o mpileup.o mpileup.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o bam2bcf.o bam2bcf.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o bam2bcf_indel.o bam2bcf_indel.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o bam2bcf_iaux.o bam2bcf_iaux.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o read_consensus.o read_consensus.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o bam_sample.o bam_sample.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o vcfsort.o vcfsort.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o cols.o cols.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o extsort.o extsort.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o dist.o dist.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o abuf.o abuf.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o ccall.o ccall.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o em.o em.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o prob1.o prob1.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o kmin.o kmin.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o str_finder.o str_finder.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o gff.o gff.c\n",
            "gcc -g -Wall -O2 -I. -I../htslib  -DPLUGINPATH='\"/usr/local/libexec/bcftools\"' -c -o vcfplugin.o vcfplugin.c\n",
            "cd ../htslib && make config.h\n",
            "make[1]: Entering directory '/content/htslib'\n",
            "echo '/* Default config.h generated by Makefile */' > config.h\n",
            "echo '#ifndef _XOPEN_SOURCE' >> config.h\n",
            "echo '#define _XOPEN_SOURCE 600' >> config.h\n",
            "echo '#endif' >> config.h\n",
            "echo '#define HAVE_LIBBZ2 1' >> config.h\n",
            "echo '#define HAVE_LIBLZMA 1' >> config.h\n",
            "echo '#ifndef __APPLE__' >> config.h\n",
            "echo '#define HAVE_LZMA_H 1' >> config.h\n",
            "echo '#endif' >> config.h\n",
            "echo '#define HAVE_DRAND48 1' >> config.h\n",
            "echo '#define HAVE_LIBCURL 1' >> config.h\n",
            "if [ \"x1\" != \"x\" ]; then \\\n",
            "    echo '#define HAVE_POPCNT 1' >> config.h ; \\\n",
            "    echo '#define HAVE_SSE4_1 1' >> config.h ; \\\n",
            "    echo '#define HAVE_SSSE3 1' >> config.h ; \\\n",
            "    echo '#if defined(HTS_ALLOW_UNALIGNED) && HTS_ALLOW_UNALIGNED == 0' >> config.h ; \\\n",
            "    echo '#define UBSAN 1' >> config.h ; \\\n",
            "    echo '#endif' >> config.h ; \\\n",
            "fi\n",
            "if [ \"x1\" != \"x\" ] ; then \\\n",
            "    echo '#define HAVE_AVX2 1' >> config.h ; \\\n",
            "fi\n",
            "if [ \"x1\" != \"x\" ] ; then \\\n",
            "    echo '#define HAVE_AVX512 1' >> config.h ; \\\n",
            "fi\n",
            "make[1]: Leaving directory '/content/htslib'\n",
            "cd ../htslib && make hts-object-files\n",
            "make[1]: Entering directory '/content/htslib'\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o kfunc.o kfunc.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o kstring.o kstring.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o bcf_sr_sort.o bcf_sr_sort.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o bgzf.o bgzf.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o errmod.o errmod.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o faidx.o faidx.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o header.o header.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o hfile.o hfile.c\n",
            "echo '#define HTS_VERSION_TEXT \"1.19.1-1-g65ae5744\"' > version.h\n",
            "printf '#define HTS_CC \"%s\"\\n#define HTS_CPPFLAGS \"%s\"\\n#define HTS_CFLAGS \"%s\"\\n#define HTS_LDFLAGS \"%s\"\\n#define HTS_LIBS \"%s\"\\n' \\\n",
            "'gcc' \\\n",
            "'' \\\n",
            "'-g -Wall -O2 -fvisibility=hidden' \\\n",
            "'-fvisibility=hidden' \\\n",
            "'-lz -lm -lbz2 -llzma -lcurl' > config_vars.h\n",
            "Updating htscodecs/htscodecs/version.h : #define HTSCODECS_VERSION_TEXT \"1.6.0\"\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o hts.o hts.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o hts_expr.o hts_expr.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o hts_os.o hts_os.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o md5.o md5.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o multipart.o multipart.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o probaln.o probaln.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o realn.o realn.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o regidx.o regidx.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o region.o region.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o sam.o sam.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o sam_mods.o sam_mods.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o synced_bcf_reader.o synced_bcf_reader.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o vcf_sweep.o vcf_sweep.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o tbx.o tbx.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o textutils.o textutils.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o thread_pool.o thread_pool.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o vcf.o vcf.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o vcfutils.o vcfutils.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o cram/cram_codecs.o cram/cram_codecs.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o cram/cram_decode.o cram/cram_decode.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o cram/cram_encode.o cram/cram_encode.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o cram/cram_external.o cram/cram_external.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o cram/cram_index.o cram/cram_index.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o cram/cram_io.o cram/cram_io.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o cram/cram_stats.o cram/cram_stats.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o cram/mFILE.o cram/mFILE.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o cram/open_trace_file.o cram/open_trace_file.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o cram/pooled_alloc.o cram/pooled_alloc.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o cram/string_alloc.o cram/string_alloc.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o htscodecs/htscodecs/arith_dynamic.o htscodecs/htscodecs/arith_dynamic.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o htscodecs/htscodecs/fqzcomp_qual.o htscodecs/htscodecs/fqzcomp_qual.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o htscodecs/htscodecs/htscodecs.o htscodecs/htscodecs/htscodecs.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o htscodecs/htscodecs/pack.o htscodecs/htscodecs/pack.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o htscodecs/htscodecs/rANS_static4x16pr.o htscodecs/htscodecs/rANS_static4x16pr.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden -mavx2 -mpopcnt -I.  -c -o htscodecs/htscodecs/rANS_static32x16pr_avx2.o htscodecs/htscodecs/rANS_static32x16pr_avx2.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden -mavx512f -mpopcnt -I.  -c -o htscodecs/htscodecs/rANS_static32x16pr_avx512.o htscodecs/htscodecs/rANS_static32x16pr_avx512.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden -msse4.1 -mpopcnt -mssse3 -I.  -c -o htscodecs/htscodecs/rANS_static32x16pr_sse4.o htscodecs/htscodecs/rANS_static32x16pr_sse4.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o htscodecs/htscodecs/rANS_static32x16pr_neon.o htscodecs/htscodecs/rANS_static32x16pr_neon.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o htscodecs/htscodecs/rANS_static32x16pr.o htscodecs/htscodecs/rANS_static32x16pr.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o htscodecs/htscodecs/rANS_static.o htscodecs/htscodecs/rANS_static.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o htscodecs/htscodecs/rle.o htscodecs/htscodecs/rle.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o htscodecs/htscodecs/tokenise_name3.o htscodecs/htscodecs/tokenise_name3.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o htscodecs/htscodecs/utils.o htscodecs/htscodecs/utils.c\n",
            "gcc -g -Wall -O2 -fvisibility=hidden  -I.  -c -o hfile_libcurl.o hfile_libcurl.c\n",
            "touch hts-object-files\n",
            "make[1]: Leaving directory '/content/htslib'\n",
            "cd ../htslib && make lib-static\n",
            "make[1]: Entering directory '/content/htslib'\n",
            "ar -rc libhts.a kfunc.o kstring.o bcf_sr_sort.o bgzf.o errmod.o faidx.o header.o hfile.o hts.o hts_expr.o hts_os.o md5.o multipart.o probaln.o realn.o regidx.o region.o sam.o sam_mods.o synced_bcf_reader.o vcf_sweep.o tbx.o textutils.o thread_pool.o vcf.o vcfutils.o cram/cram_codecs.o cram/cram_decode.o cram/cram_encode.o cram/cram_external.o cram/cram_index.o cram/cram_io.o cram/cram_stats.o cram/mFILE.o cram/open_trace_file.o cram/pooled_alloc.o cram/string_alloc.o htscodecs/htscodecs/arith_dynamic.o htscodecs/htscodecs/fqzcomp_qual.o htscodecs/htscodecs/htscodecs.o htscodecs/htscodecs/pack.o htscodecs/htscodecs/rANS_static4x16pr.o htscodecs/htscodecs/rANS_static32x16pr_avx2.o htscodecs/htscodecs/rANS_static32x16pr_avx512.o htscodecs/htscodecs/rANS_static32x16pr_sse4.o htscodecs/htscodecs/rANS_static32x16pr_neon.o htscodecs/htscodecs/rANS_static32x16pr.o htscodecs/htscodecs/rANS_static.o htscodecs/htscodecs/rle.o htscodecs/htscodecs/tokenise_name3.o htscodecs/htscodecs/utils.o hfile_libcurl.o\n",
            "ranlib libhts.a\n",
            "make[1]: Leaving directory '/content/htslib'\n",
            "gcc -rdynamic   -o bcftools main.o vcfindex.o tabix.o vcfstats.o vcfisec.o vcfmerge.o vcfquery.o vcffilter.o filter.o vcfsom.o vcfnorm.o vcfgtcheck.o vcfview.o vcfannotate.o vcfroh.o vcfconcat.o vcfcall.o mcall.o vcmp.o gvcf.o reheader.o convert.o vcfconvert.o tsv2vcf.o vcfcnv.o vcfhead.o HMM.o consensus.o ploidy.o bin.o hclust.o version.o regidx.o smpl_ilist.o csq.o vcfbuf.o mpileup.o bam2bcf.o bam2bcf_indel.o bam2bcf_iaux.o read_consensus.o bam_sample.o vcfsort.o cols.o extsort.o dist.o abuf.o ccall.o em.o prob1.o kmin.o str_finder.o gff.o vcfplugin.o ../htslib/libhts.a -lz -lm -lbz2 -llzma -lcurl -lm -lz -ldl    -lpthread\n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o test/test-rbuf.o test/test-rbuf.c\n",
            "gcc  -o test/test-rbuf test/test-rbuf.o -lz -ldl \n",
            "gcc -g -Wall -O2 -I. -I../htslib   -c -o test/test-regidx.o test/test-regidx.c\n",
            "gcc   -o test/test-regidx test/test-regidx.o regidx.o ../htslib/libhts.a -lz -lm -lbz2 -llzma -lcurl -lpthread -lz -ldl \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/ad-bias.so convert.c version.c plugins/ad-bias.c  \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/add-variantkey.so version.c plugins/add-variantkey.c \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/af-dist.so bin.c version.c plugins/af-dist.c  \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/allele-length.so version.c plugins/allele-length.c \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/check-ploidy.so version.c plugins/check-ploidy.c \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/check-sparsity.so version.c plugins/check-sparsity.c \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/color-chrs.so HMM.c version.c plugins/color-chrs.c  \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/contrast.so version.c plugins/contrast.c \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/counts.so version.c plugins/counts.c \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/dosage.so version.c plugins/dosage.c \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/fill-AN-AC.so version.c plugins/fill-AN-AC.c \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib     -o plugins/fill-from-fasta.so filter.c version.c plugins/fill-from-fasta.c  \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/fill-tags.so version.c plugins/fill-tags.c \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/fixploidy.so ploidy.c regidx.c version.c plugins/fixploidy.c  \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/fixref.so version.c plugins/fixref.c \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/frameshifts.so version.c plugins/frameshifts.c \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/GTisec.so version.c plugins/GTisec.c  \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/GTsubset.so version.c plugins/GTsubset.c  \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/guess-ploidy.so version.c plugins/guess-ploidy.c \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/gvcfz.so version.c plugins/gvcfz.c \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/impute-info.so version.c plugins/impute-info.c \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/indel-stats.so version.c plugins/indel-stats.c \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/isecGT.so smpl_ilist.c version.c plugins/isecGT.c  \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/mendelian2.so version.c plugins/mendelian2.c \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/missing2ref.so version.c plugins/missing2ref.c \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/parental-origin.so version.c plugins/parental-origin.c \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/prune.so version.c plugins/prune.c \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/remove-overlaps.so version.c plugins/remove-overlaps.c \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/scatter.so version.c plugins/scatter.c \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/setGT.so version.c plugins/setGT.c \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/smpl-stats.so version.c plugins/smpl-stats.c \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/split.so version.c plugins/split.c \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/split-vep.so version.c plugins/split-vep.c \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/tag2tag.so version.c plugins/tag2tag.c \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/trio-dnm2.so version.c plugins/trio-dnm2.c \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/trio-stats.so version.c plugins/trio-stats.c \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/trio-switch-rate.so version.c plugins/trio-switch-rate.c \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/variant-distance.so version.c plugins/variant-distance.c \n",
            "gcc -fPIC -shared -g -Wall -O2 -I. -I../htslib    -o plugins/variantkey-hex.so version.c plugins/variantkey-hex.c \n",
            "mkdir -p -m 755 /usr/local/bin /usr/local/share/man/man1 /usr/local/libexec/bcftools\n",
            "install -p bcftools /usr/local/bin\n",
            "install -p misc/color-chrs.pl misc/gff2gff.py misc/guess-ploidy.py misc/plot-vcfstats misc/plot-roh.py misc/run-roh.pl misc/vcfutils.pl /usr/local/bin\n",
            "install -p -m 644 doc/bcftools.1 /usr/local/share/man/man1\n",
            "install -p plugins/*.so /usr/local/libexec/bcftools\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "Cloning into 'htslib'...\n",
            "Submodule 'htscodecs' (https://github.com/samtools/htscodecs.git) registered for path 'htscodecs'\n",
            "Cloning into '/content/htslib/htscodecs'...\n",
            "Cloning into 'bcftools'...\n"
          ]
        }
      ],
      "source": [
        "#permite extrair os campos de anotações estruturadas como INFO/CSQ criadas por VEP\n",
        "%%bash\n",
        "git clone --recurse-submodules https://github.com/samtools/htslib.git\n",
        "git clone https://github.com/samtools/bcftools.git\n",
        "cd bcftools\n",
        "make\n",
        "make install"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "hBIBl0xvLt_D"
      },
      "source": [
        "## 3. Instalar udocker\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "ymnqB8Uca1vn",
        "outputId": "70b6ddec-b0ec-4d92-892e-6bf777c6133f"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting udocker\n",
            "  Downloading udocker-1.3.12-py2.py3-none-any.whl (118 kB)\n",
            "     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 118.0/118.0 kB 1.4 MB/s eta 0:00:00\n",
            "Installing collected packages: udocker\n",
            "Successfully installed udocker-1.3.12\n",
            "Info: creating repo: /root/.udocker\n",
            "Info: udocker command line interface 1.3.12\n",
            "Info: searching for udockertools >= 1.2.10\n",
            "Info: installing udockertools 1.2.10\n",
            "Info: installation of udockertools successful\n"
          ]
        }
      ],
      "source": [
        "%%bash\n",
        "# Fonte: https://gist.github.com/mwufi/6718b30761cd109f9aff04c5144eb885\n",
        "pip install udocker\n",
        "udocker --allow-root install"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "Zotlwj4_MJ4l"
      },
      "source": [
        "## 4. Baixar imagem do ensembl-vep\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "YoEPm3j2-x4L",
        "outputId": "f60c168d-3242-455a-9f31-6f8ef977e130"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Info: downloading layer sha256:d57d4eb4e8d71e1a3b27b0515d9ffb6f053291c0bf4667f22980dcfe84b6913a\n",
            "Info: downloading layer sha256:8da5acaae4c91044ca467ac4bb74e41badcce7ed0e2e7f12e2d6c2a3f7333b6d\n",
            "Info: downloading layer sha256:8a144600929dc8a2519022c13ef41d984920440d53af8b2736b928e56465eb16\n",
            "Info: downloading layer sha256:e9c96579a6c7fc8199cd61ac60ea4fc2e083ef03e470ac5b9e0dabfd1f52adb8\n",
            "Info: downloading layer sha256:63ae9cf8caa98c5e9c7695631faa795cd06a0debda97a55baa832625a52b0b74\n",
            "Info: downloading layer sha256:50045e6409ebc1687df4548cfc330f1fad340dfa92b7a84ab3985da7642f5a71\n",
            "Info: downloading layer sha256:a1ec7562e9cd6a332e524d590b8b20413457228487629b4a8f155a121f406ba2\n",
            "Info: downloading layer sha256:a2230cfa06a7b5a218b0d0f23d53cd3af410839a6ed6ab9883502611ec9d321b\n",
            "Info: downloading layer sha256:cebaa0fecec3b5b9a48b37aed9cc625da104156617c497e0440a03944ae7dc5f\n",
            "Info: downloading layer sha256:4f4fb700ef54461cfa02571ae0db9a0dc1e0cdb5577484a6d75e68dc38e8acc1\n",
            "Info: downloading layer sha256:9bec7112ad258d751f29d4acd8ead0f1bfc0823eb09496a150d60d5b710e92f4\n",
            "Info: downloading layer sha256:1ce38d1f7f17b4baa519c172236689deb82fc04ea06fb47cafab3ac87b149d0e\n",
            "Info: downloading layer sha256:4f4fb700ef54461cfa02571ae0db9a0dc1e0cdb5577484a6d75e68dc38e8acc1\n",
            "Info: downloading layer sha256:3b6feca4f0109f08c266223b5a94524a8b176405b379f52f36c6c1cdb17dba6f\n",
            "Info: downloading layer sha256:4f4fb700ef54461cfa02571ae0db9a0dc1e0cdb5577484a6d75e68dc38e8acc1\n",
            "Info: downloading layer sha256:dbd44792cf40056af4d1f3214bc484584b085a07b4b776cac08220f3f0d6f98a\n",
            "Info: downloading layer sha256:4f4fb700ef54461cfa02571ae0db9a0dc1e0cdb5577484a6d75e68dc38e8acc1\n",
            "Info: downloading layer sha256:0ba352db6b1e187b01f1e90c1c434c6f06266ee18419a6f4432f132d617170d7\n",
            "Info: downloading layer sha256:7f1f0e3c5a4ab3b1bb2b50ae46a0eda32e80c25dfa777fe6ea4f8388ae3f00ea\n",
            "Info: downloading layer sha256:566e2c264d75e7b17f7be1fd02b16443f5e86c541f79679f7a46f41fe5967103\n",
            "Info: downloading layer sha256:29202e855b2021a2d7f92800619ed5f5e8ac402e267cfbb3d29a791feb13c1ee\n"
          ]
        }
      ],
      "source": [
        "!udocker --allow-root pull ensemblorg/ensembl-vep"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "ivMMnCnEMz9S"
      },
      "source": [
        "## 5. Filtragem"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "V23tjUzPGJCY"
      },
      "outputs": [],
      "source": [
        "#Declaração de lista com nome das 30 amostras do projeto lma brasil\n",
        "SAMPLES = [\"WP048\",\"WP093\",\"WP087\",\"WP060\",\"WP056\",\"WP066\",\"WP064\",\"WP072\",\"WP078\",\"WP285\",\"WP280\",\"WP274\",\"WP276\",\"WP270\",\"WP216\",\"WP306\",\"WP297\",\"WP291\",\"WP295\",\"WP204\",\"WP160\",\"WP164\",\"WP162\",\"WP212\",\"WP170\",\"WP196\",\"WP180\",\"WP188\",\"WP140\",\"WP126\"]"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "p8i8WTPwxtV8",
        "outputId": "c4feee42-dc04-4cb9-88eb-390b0acf84b5"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "WP048\n",
            " \n",
            " ****************************************************************************** \n",
            " *                                                                            * \n",
            " *               STARTING 3adea43c-b9dd-3d24-8f31-4ac88234a918                * \n",
            " *                                                                            * \n",
            " ****************************************************************************** \n",
            " executing: filter_vep\n",
            "\n",
            "-------------------- WARNING ----------------------\n",
            "MSG: The script 'filter_vep' couldn't find the VEP annotations INFO field CSQ in 149 lines of the input file\n",
            "FILE: src/ensembl-vep/filter_vep LINE: 345\n",
            "CALLED BY: src/ensembl-vep/filter_vep  LINE: 55\n",
            "Date (localtime)    = Sat Jan 27 22:10:43 2024\n",
            "Ensembl API version = 111\n",
            "---------------------------------------------------\n",
            "WP093\n",
            " \n",
            " ****************************************************************************** \n",
            " *                                                                            * \n",
            " *               STARTING 21174e80-c520-364a-bd1d-ee27b7b1698d                * \n",
            " *                                                                            * \n",
            " ****************************************************************************** \n",
            " executing: filter_vep\n",
            "\n",
            "-------------------- WARNING ----------------------\n",
            "MSG: The script 'filter_vep' couldn't find the VEP annotations INFO field CSQ in 188 lines of the input file\n",
            "FILE: src/ensembl-vep/filter_vep LINE: 345\n",
            "CALLED BY: src/ensembl-vep/filter_vep  LINE: 55\n",
            "Date (localtime)    = Sat Jan 27 22:11:13 2024\n",
            "Ensembl API version = 111\n",
            "---------------------------------------------------\n",
            "Warning: Could not parse, not a numeric list xgnomADg_AF=\".\", check the -c and --columns-types options.\n",
            "         This message is printed only once.\n",
            "WP087\n",
            " \n",
            " ****************************************************************************** \n",
            " *                                                                            * \n",
            " *               STARTING 6fd5a377-f600-3463-a354-6ec4163d680b                * \n",
            " *                                                                            * \n",
            " ****************************************************************************** \n",
            " executing: filter_vep\n",
            "\n",
            "-------------------- WARNING ----------------------\n",
            "MSG: The script 'filter_vep' couldn't find the VEP annotations INFO field CSQ in 132 lines of the input file\n",
            "FILE: src/ensembl-vep/filter_vep LINE: 345\n",
            "CALLED BY: src/ensembl-vep/filter_vep  LINE: 55\n",
            "Date (localtime)    = Sat Jan 27 22:11:43 2024\n",
            "Ensembl API version = 111\n",
            "---------------------------------------------------\n",
            "WP060\n",
            " \n",
            " ****************************************************************************** \n",
            " *                                                                            * \n",
            " *               STARTING 9ed09871-324a-3f5c-be3d-6dc72ddd8985                * \n",
            " *                                                                            * \n",
            " ****************************************************************************** \n",
            " executing: filter_vep\n",
            "\n",
            "-------------------- WARNING ----------------------\n",
            "MSG: The script 'filter_vep' couldn't find the VEP annotations INFO field CSQ in 146 lines of the input file\n",
            "FILE: src/ensembl-vep/filter_vep LINE: 345\n",
            "CALLED BY: src/ensembl-vep/filter_vep  LINE: 55\n",
            "Date (localtime)    = Sat Jan 27 22:12:14 2024\n",
            "Ensembl API version = 111\n",
            "---------------------------------------------------\n",
            "Warning: Could not parse, not a numeric list xgnomADg_AF=\"9.206e-05\", check the -c and --columns-types options.\n",
            "         This message is printed only once.\n",
            "WP056\n",
            " \n",
            " ****************************************************************************** \n",
            " *                                                                            * \n",
            " *               STARTING df2d29c7-17ef-3752-ab63-6285e981dde6                * \n",
            " *                                                                            * \n",
            " ****************************************************************************** \n",
            " executing: filter_vep\n",
            "\n",
            "-------------------- WARNING ----------------------\n",
            "MSG: The script 'filter_vep' couldn't find the VEP annotations INFO field CSQ in 133 lines of the input file\n",
            "FILE: src/ensembl-vep/filter_vep LINE: 345\n",
            "CALLED BY: src/ensembl-vep/filter_vep  LINE: 55\n",
            "Date (localtime)    = Sat Jan 27 22:12:44 2024\n",
            "Ensembl API version = 111\n",
            "---------------------------------------------------\n",
            "Warning: Could not parse, not a numeric list xgnomADg_AF=\"0.0001127\", check the -c and --columns-types options.\n",
            "         This message is printed only once.\n",
            "WP066\n",
            " \n",
            " ****************************************************************************** \n",
            " *                                                                            * \n",
            " *               STARTING a4967656-4bd1-33fd-91a5-3f66d3a68e8f                * \n",
            " *                                                                            * \n",
            " ****************************************************************************** \n",
            " executing: filter_vep\n",
            "\n",
            "-------------------- WARNING ----------------------\n",
            "MSG: The script 'filter_vep' couldn't find the VEP annotations INFO field CSQ in 149 lines of the input file\n",
            "FILE: src/ensembl-vep/filter_vep LINE: 345\n",
            "CALLED BY: src/ensembl-vep/filter_vep  LINE: 55\n",
            "Date (localtime)    = Sat Jan 27 22:13:12 2024\n",
            "Ensembl API version = 111\n",
            "---------------------------------------------------\n",
            "WP064\n",
            " \n",
            " ****************************************************************************** \n",
            " *                                                                            * \n",
            " *               STARTING e4fed420-9ea1-3a8e-bc00-703098f8136d                * \n",
            " *                                                                            * \n",
            " ****************************************************************************** \n",
            " executing: filter_vep\n",
            "\n",
            "-------------------- WARNING ----------------------\n",
            "MSG: The script 'filter_vep' couldn't find the VEP annotations INFO field CSQ in 125 lines of the input file\n",
            "FILE: src/ensembl-vep/filter_vep LINE: 345\n",
            "CALLED BY: src/ensembl-vep/filter_vep  LINE: 55\n",
            "Date (localtime)    = Sat Jan 27 22:13:43 2024\n",
            "Ensembl API version = 111\n",
            "---------------------------------------------------\n",
            "WP072\n",
            " \n",
            " ****************************************************************************** \n",
            " *                                                                            * \n",
            " *               STARTING 70936681-f556-3a21-ab7d-4824b5b1f49b                * \n",
            " *                                                                            * \n",
            " ****************************************************************************** \n",
            " executing: filter_vep\n",
            "\n",
            "-------------------- WARNING ----------------------\n",
            "MSG: The script 'filter_vep' couldn't find the VEP annotations INFO field CSQ in 221 lines of the input file\n",
            "FILE: src/ensembl-vep/filter_vep LINE: 345\n",
            "CALLED BY: src/ensembl-vep/filter_vep  LINE: 55\n",
            "Date (localtime)    = Sat Jan 27 22:14:13 2024\n",
            "Ensembl API version = 111\n",
            "---------------------------------------------------\n",
            "WP078\n",
            " \n",
            " ****************************************************************************** \n",
            " *                                                                            * \n",
            " *               STARTING a432682d-cbd5-3a88-9b2c-973ae383ee82                * \n",
            " *                                                                            * \n",
            " ****************************************************************************** \n",
            " executing: filter_vep\n",
            "\n",
            "-------------------- WARNING ----------------------\n",
            "MSG: The script 'filter_vep' couldn't find the VEP annotations INFO field CSQ in 170 lines of the input file\n",
            "FILE: src/ensembl-vep/filter_vep LINE: 345\n",
            "CALLED BY: src/ensembl-vep/filter_vep  LINE: 55\n",
            "Date (localtime)    = Sat Jan 27 22:14:41 2024\n",
            "Ensembl API version = 111\n",
            "---------------------------------------------------\n",
            "Warning: Could not parse, not a numeric list xgnomADg_AF=\"1.971e-05\", check the -c and --columns-types options.\n",
            "         This message is printed only once.\n",
            "WP285\n",
            " \n",
            " ****************************************************************************** \n",
            " *                                                                            * \n",
            " *               STARTING 41ce0f1b-4711-3c90-b96a-5981b923383f                * \n",
            " *                                                                            * \n",
            " ****************************************************************************** \n",
            " executing: filter_vep\n",
            "\n",
            "-------------------- WARNING ----------------------\n",
            "MSG: The script 'filter_vep' couldn't find the VEP annotations INFO field CSQ in 191 lines of the input file\n",
            "FILE: src/ensembl-vep/filter_vep LINE: 345\n",
            "CALLED BY: src/ensembl-vep/filter_vep  LINE: 55\n",
            "Date (localtime)    = Sat Jan 27 22:15:16 2024\n",
            "Ensembl API version = 111\n",
            "---------------------------------------------------\n",
            "Warning: Could not parse, not a numeric list xgnomADg_AF=\".\", check the -c and --columns-types options.\n",
            "         This message is printed only once.\n",
            "WP280\n",
            " \n",
            " ****************************************************************************** \n",
            " *                                                                            * \n",
            " *               STARTING 8397e797-19d4-33b8-a2da-63bcfb31ed0d                * \n",
            " *                                                                            * \n",
            " ****************************************************************************** \n",
            " executing: filter_vep\n",
            "\n",
            "-------------------- WARNING ----------------------\n",
            "MSG: The script 'filter_vep' couldn't find the VEP annotations INFO field CSQ in 180 lines of the input file\n",
            "FILE: src/ensembl-vep/filter_vep LINE: 345\n",
            "CALLED BY: src/ensembl-vep/filter_vep  LINE: 55\n",
            "Date (localtime)    = Sat Jan 27 22:15:48 2024\n",
            "Ensembl API version = 111\n",
            "---------------------------------------------------\n",
            "WP274\n",
            " \n",
            " ****************************************************************************** \n",
            " *                                                                            * \n",
            " *               STARTING 58053e99-7a25-3ec2-91da-1462bf467a7a                * \n",
            " *                                                                            * \n",
            " ****************************************************************************** \n",
            " executing: filter_vep\n",
            "\n",
            "-------------------- WARNING ----------------------\n",
            "MSG: The script 'filter_vep' couldn't find the VEP annotations INFO field CSQ in 136 lines of the input file\n",
            "FILE: src/ensembl-vep/filter_vep LINE: 345\n",
            "CALLED BY: src/ensembl-vep/filter_vep  LINE: 55\n",
            "Date (localtime)    = Sat Jan 27 22:16:19 2024\n",
            "Ensembl API version = 111\n",
            "---------------------------------------------------\n",
            "Warning: Could not parse, not a numeric list xgnomADg_AF=\"1.971e-05\", check the -c and --columns-types options.\n",
            "         This message is printed only once.\n",
            "WP276\n",
            " \n",
            " ****************************************************************************** \n",
            " *                                                                            * \n",
            " *               STARTING 7b3b5a17-2f71-317d-ad54-7f76d2e5f246                * \n",
            " *                                                                            * \n",
            " ****************************************************************************** \n",
            " executing: filter_vep\n",
            "\n",
            "-------------------- WARNING ----------------------\n",
            "MSG: The script 'filter_vep' couldn't find the VEP annotations INFO field CSQ in 176 lines of the input file\n",
            "FILE: src/ensembl-vep/filter_vep LINE: 345\n",
            "CALLED BY: src/ensembl-vep/filter_vep  LINE: 55\n",
            "Date (localtime)    = Sat Jan 27 22:16:50 2024\n",
            "Ensembl API version = 111\n",
            "---------------------------------------------------\n",
            "Warning: Could not parse, not a numeric list xgnomADg_AF=\"1.971e-05\", check the -c and --columns-types options.\n",
            "         This message is printed only once.\n",
            "WP270\n",
            " \n",
            " ****************************************************************************** \n",
            " *                                                                            * \n",
            " *               STARTING d77659b6-4b88-378e-8b46-d0844cf20444                * \n",
            " *                                                                            * \n",
            " ****************************************************************************** \n",
            " executing: filter_vep\n",
            "\n",
            "-------------------- WARNING ----------------------\n",
            "MSG: The script 'filter_vep' couldn't find the VEP annotations INFO field CSQ in 187 lines of the input file\n",
            "FILE: src/ensembl-vep/filter_vep LINE: 345\n",
            "CALLED BY: src/ensembl-vep/filter_vep  LINE: 55\n",
            "Date (localtime)    = Sat Jan 27 22:17:21 2024\n",
            "Ensembl API version = 111\n",
            "---------------------------------------------------\n",
            "Warning: Could not parse, not a numeric list xgnomADg_AF=\"1.971e-05\", check the -c and --columns-types options.\n",
            "         This message is printed only once.\n",
            "WP216\n",
            " \n",
            " ****************************************************************************** \n",
            " *                                                                            * \n",
            " *               STARTING e7420855-44b4-36ad-a7c7-af43cffd6675                * \n",
            " *                                                                            * \n",
            " ****************************************************************************** \n",
            " executing: filter_vep\n",
            "\n",
            "-------------------- WARNING ----------------------\n",
            "MSG: The script 'filter_vep' couldn't find the VEP annotations INFO field CSQ in 209 lines of the input file\n",
            "FILE: src/ensembl-vep/filter_vep LINE: 345\n",
            "CALLED BY: src/ensembl-vep/filter_vep  LINE: 55\n",
            "Date (localtime)    = Sat Jan 27 22:17:52 2024\n",
            "Ensembl API version = 111\n",
            "---------------------------------------------------\n",
            "Warning: Could not parse, not a numeric list xgnomADg_AF=\"0\", check the -c and --columns-types options.\n",
            "         This message is printed only once.\n",
            "WP306\n",
            " \n",
            " ****************************************************************************** \n",
            " *                                                                            * \n",
            " *               STARTING 5177bc13-b58c-3f4b-9a80-8b57a7002af2                * \n",
            " *                                                                            * \n",
            " ****************************************************************************** \n",
            " executing: filter_vep\n",
            "\n",
            "-------------------- WARNING ----------------------\n",
            "MSG: The script 'filter_vep' couldn't find the VEP annotations INFO field CSQ in 452 lines of the input file\n",
            "FILE: src/ensembl-vep/filter_vep LINE: 345\n",
            "CALLED BY: src/ensembl-vep/filter_vep  LINE: 55\n",
            "Date (localtime)    = Sat Jan 27 22:18:29 2024\n",
            "Ensembl API version = 111\n",
            "---------------------------------------------------\n",
            "Warning: Could not parse, not a numeric list xgnomADg_AF=\"1.972e-05\", check the -c and --columns-types options.\n",
            "         This message is printed only once.\n",
            "WP297\n",
            " \n",
            " ****************************************************************************** \n",
            " *                                                                            * \n",
            " *               STARTING 3cc9a6f8-4934-3a85-a4b1-53f69a2e0fea                * \n",
            " *                                                                            * \n",
            " ****************************************************************************** \n",
            " executing: filter_vep\n",
            "\n",
            "-------------------- WARNING ----------------------\n",
            "MSG: The script 'filter_vep' couldn't find the VEP annotations INFO field CSQ in 155 lines of the input file\n",
            "FILE: src/ensembl-vep/filter_vep LINE: 345\n",
            "CALLED BY: src/ensembl-vep/filter_vep  LINE: 55\n",
            "Date (localtime)    = Sat Jan 27 22:18:58 2024\n",
            "Ensembl API version = 111\n",
            "---------------------------------------------------\n",
            "Warning: Could not parse, not a numeric list xgnomADg_AF=\"1.971e-05\", check the -c and --columns-types options.\n",
            "         This message is printed only once.\n",
            "WP291\n",
            " \n",
            " ****************************************************************************** \n",
            " *                                                                            * \n",
            " *               STARTING f5cc4bd6-60dd-3770-99ec-697e5fdf436f                * \n",
            " *                                                                            * \n",
            " ****************************************************************************** \n",
            " executing: filter_vep\n",
            "\n",
            "-------------------- WARNING ----------------------\n",
            "MSG: The script 'filter_vep' couldn't find the VEP annotations INFO field CSQ in 179 lines of the input file\n",
            "FILE: src/ensembl-vep/filter_vep LINE: 345\n",
            "CALLED BY: src/ensembl-vep/filter_vep  LINE: 55\n",
            "Date (localtime)    = Sat Jan 27 22:19:29 2024\n",
            "Ensembl API version = 111\n",
            "---------------------------------------------------\n",
            "WP295\n",
            " \n",
            " ****************************************************************************** \n",
            " *                                                                            * \n",
            " *               STARTING c75e02f3-4990-3695-b886-b82dd88d925a                * \n",
            " *                                                                            * \n",
            " ****************************************************************************** \n",
            " executing: filter_vep\n",
            "\n",
            "-------------------- WARNING ----------------------\n",
            "MSG: The script 'filter_vep' couldn't find the VEP annotations INFO field CSQ in 276 lines of the input file\n",
            "FILE: src/ensembl-vep/filter_vep LINE: 345\n",
            "CALLED BY: src/ensembl-vep/filter_vep  LINE: 55\n",
            "Date (localtime)    = Sat Jan 27 22:20:02 2024\n",
            "Ensembl API version = 111\n",
            "---------------------------------------------------\n",
            "Warning: Could not parse, not a numeric list xgnomADg_AF=\".\", check the -c and --columns-types options.\n",
            "         This message is printed only once.\n",
            "WP204\n",
            " \n",
            " ****************************************************************************** \n",
            " *                                                                            * \n",
            " *               STARTING d930251c-0974-3279-9364-72ef28435e06                * \n",
            " *                                                                            * \n",
            " ****************************************************************************** \n",
            " executing: filter_vep\n",
            "\n",
            "-------------------- WARNING ----------------------\n",
            "MSG: The script 'filter_vep' couldn't find the VEP annotations INFO field CSQ in 192 lines of the input file\n",
            "FILE: src/ensembl-vep/filter_vep LINE: 345\n",
            "CALLED BY: src/ensembl-vep/filter_vep  LINE: 55\n",
            "Date (localtime)    = Sat Jan 27 22:20:35 2024\n",
            "Ensembl API version = 111\n",
            "---------------------------------------------------\n",
            "WP160\n",
            " \n",
            " ****************************************************************************** \n",
            " *                                                                            * \n",
            " *               STARTING 6c52b4db-54d8-3fac-a85d-7f53e1f5f203                * \n",
            " *                                                                            * \n",
            " ****************************************************************************** \n",
            " executing: filter_vep\n",
            "\n",
            "-------------------- WARNING ----------------------\n",
            "MSG: The script 'filter_vep' couldn't find the VEP annotations INFO field CSQ in 119 lines of the input file\n",
            "FILE: src/ensembl-vep/filter_vep LINE: 345\n",
            "CALLED BY: src/ensembl-vep/filter_vep  LINE: 55\n",
            "Date (localtime)    = Sat Jan 27 22:21:06 2024\n",
            "Ensembl API version = 111\n",
            "---------------------------------------------------\n",
            "Warning: Could not parse, not a numeric list xgnomADg_AF=\"1.971e-05\", check the -c and --columns-types options.\n",
            "         This message is printed only once.\n",
            "WP164\n",
            " \n",
            " ****************************************************************************** \n",
            " *                                                                            * \n",
            " *               STARTING a5039e75-08b1-30e7-b1d1-e964c3b9e220                * \n",
            " *                                                                            * \n",
            " ****************************************************************************** \n",
            " executing: filter_vep\n",
            "\n",
            "-------------------- WARNING ----------------------\n",
            "MSG: The script 'filter_vep' couldn't find the VEP annotations INFO field CSQ in 214 lines of the input file\n",
            "FILE: src/ensembl-vep/filter_vep LINE: 345\n",
            "CALLED BY: src/ensembl-vep/filter_vep  LINE: 55\n",
            "Date (localtime)    = Sat Jan 27 22:21:36 2024\n",
            "Ensembl API version = 111\n",
            "---------------------------------------------------\n",
            "WP162\n",
            " \n",
            " ****************************************************************************** \n",
            " *                                                                            * \n",
            " *               STARTING 964fbc6c-8af1-3ad4-a598-03947f8e346b                * \n",
            " *                                                                            * \n",
            " ****************************************************************************** \n",
            " executing: filter_vep\n",
            "\n",
            "-------------------- WARNING ----------------------\n",
            "MSG: The script 'filter_vep' couldn't find the VEP annotations INFO field CSQ in 87 lines of the input file\n",
            "FILE: src/ensembl-vep/filter_vep LINE: 345\n",
            "CALLED BY: src/ensembl-vep/filter_vep  LINE: 55\n",
            "Date (localtime)    = Sat Jan 27 22:22:06 2024\n",
            "Ensembl API version = 111\n",
            "---------------------------------------------------\n",
            "WP212\n",
            " \n",
            " ****************************************************************************** \n",
            " *                                                                            * \n",
            " *               STARTING 393af72e-d3b3-3461-a4b0-16dae5180396                * \n",
            " *                                                                            * \n",
            " ****************************************************************************** \n",
            " executing: filter_vep\n",
            "\n",
            "-------------------- WARNING ----------------------\n",
            "MSG: The script 'filter_vep' couldn't find the VEP annotations INFO field CSQ in 193 lines of the input file\n",
            "FILE: src/ensembl-vep/filter_vep LINE: 345\n",
            "CALLED BY: src/ensembl-vep/filter_vep  LINE: 55\n",
            "Date (localtime)    = Sat Jan 27 22:22:36 2024\n",
            "Ensembl API version = 111\n",
            "---------------------------------------------------\n",
            "Warning: Could not parse, not a numeric list xgnomADg_AF=\"0.0001184\", check the -c and --columns-types options.\n",
            "         This message is printed only once.\n",
            "WP170\n",
            " \n",
            " ****************************************************************************** \n",
            " *                                                                            * \n",
            " *               STARTING 99fe2f76-6091-3c27-ab1a-5267c05b86ef                * \n",
            " *                                                                            * \n",
            " ****************************************************************************** \n",
            " executing: filter_vep\n",
            "\n",
            "-------------------- WARNING ----------------------\n",
            "MSG: The script 'filter_vep' couldn't find the VEP annotations INFO field CSQ in 150 lines of the input file\n",
            "FILE: src/ensembl-vep/filter_vep LINE: 345\n",
            "CALLED BY: src/ensembl-vep/filter_vep  LINE: 55\n",
            "Date (localtime)    = Sat Jan 27 22:23:07 2024\n",
            "Ensembl API version = 111\n",
            "---------------------------------------------------\n",
            "WP196\n",
            " \n",
            " ****************************************************************************** \n",
            " *                                                                            * \n",
            " *               STARTING 6918d19f-d025-39dd-a1a6-2c373243a4c7                * \n",
            " *                                                                            * \n",
            " ****************************************************************************** \n",
            " executing: filter_vep\n",
            "\n",
            "-------------------- WARNING ----------------------\n",
            "MSG: The script 'filter_vep' couldn't find the VEP annotations INFO field CSQ in 140 lines of the input file\n",
            "FILE: src/ensembl-vep/filter_vep LINE: 345\n",
            "CALLED BY: src/ensembl-vep/filter_vep  LINE: 55\n",
            "Date (localtime)    = Sat Jan 27 22:23:36 2024\n",
            "Ensembl API version = 111\n",
            "---------------------------------------------------\n",
            "WP180\n",
            " \n",
            " ****************************************************************************** \n",
            " *                                                                            * \n",
            " *               STARTING 52225d0c-ac9b-3740-859e-4e4f9e407709                * \n",
            " *                                                                            * \n",
            " ****************************************************************************** \n",
            " executing: filter_vep\n",
            "\n",
            "-------------------- WARNING ----------------------\n",
            "MSG: The script 'filter_vep' couldn't find the VEP annotations INFO field CSQ in 158 lines of the input file\n",
            "FILE: src/ensembl-vep/filter_vep LINE: 345\n",
            "CALLED BY: src/ensembl-vep/filter_vep  LINE: 55\n",
            "Date (localtime)    = Sat Jan 27 22:24:04 2024\n",
            "Ensembl API version = 111\n",
            "---------------------------------------------------\n",
            "WP188\n",
            " \n",
            " ****************************************************************************** \n",
            " *                                                                            * \n",
            " *               STARTING c16749b6-a3ee-3592-95a2-f00e58b0ed3f                * \n",
            " *                                                                            * \n",
            " ****************************************************************************** \n",
            " executing: filter_vep\n",
            "\n",
            "-------------------- WARNING ----------------------\n",
            "MSG: The script 'filter_vep' couldn't find the VEP annotations INFO field CSQ in 175 lines of the input file\n",
            "FILE: src/ensembl-vep/filter_vep LINE: 345\n",
            "CALLED BY: src/ensembl-vep/filter_vep  LINE: 55\n",
            "Date (localtime)    = Sat Jan 27 22:24:35 2024\n",
            "Ensembl API version = 111\n",
            "---------------------------------------------------\n",
            "Warning: Could not parse, not a numeric list xgnomADg_AF=\"1.971e-05\", check the -c and --columns-types options.\n",
            "         This message is printed only once.\n",
            "WP140\n",
            " \n",
            " ****************************************************************************** \n",
            " *                                                                            * \n",
            " *               STARTING 93ef0ef1-5de8-31fe-bd39-30084cddb9bb                * \n",
            " *                                                                            * \n",
            " ****************************************************************************** \n",
            " executing: filter_vep\n",
            "\n",
            "-------------------- WARNING ----------------------\n",
            "MSG: The script 'filter_vep' couldn't find the VEP annotations INFO field CSQ in 155 lines of the input file\n",
            "FILE: src/ensembl-vep/filter_vep LINE: 345\n",
            "CALLED BY: src/ensembl-vep/filter_vep  LINE: 55\n",
            "Date (localtime)    = Sat Jan 27 22:25:06 2024\n",
            "Ensembl API version = 111\n",
            "---------------------------------------------------\n",
            "Warning: Could not parse, not a numeric list xgnomADg_AF=\"1.971e-05\", check the -c and --columns-types options.\n",
            "         This message is printed only once.\n",
            "WP126\n",
            " \n",
            " ****************************************************************************** \n",
            " *                                                                            * \n",
            " *               STARTING 32e2c9f6-877a-393e-8a9d-c48583b47b74                * \n",
            " *                                                                            * \n",
            " ****************************************************************************** \n",
            " executing: filter_vep\n",
            "\n",
            "-------------------- WARNING ----------------------\n",
            "MSG: The script 'filter_vep' couldn't find the VEP annotations INFO field CSQ in 144 lines of the input file\n",
            "FILE: src/ensembl-vep/filter_vep LINE: 345\n",
            "CALLED BY: src/ensembl-vep/filter_vep  LINE: 55\n",
            "Date (localtime)    = Sat Jan 27 22:25:42 2024\n",
            "Ensembl API version = 111\n",
            "---------------------------------------------------\n"
          ]
        }
      ],
      "source": [
        "#loop para processar pipeline de filtragem de variantes em cada uma das 30 amostras da lista SAMPLES\n",
        "for i in SAMPLES:\n",
        "  SAMPLE = i\n",
        "  !echo {SAMPLE}\n",
        "  !sh lmabrasil-hg38/vep-gc.sh {SAMPLE} mielofibrose.txt"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "p45QBQEJGltY"
      },
      "outputs": [],
      "source": [
        "\n",
        "import pandas as pd\n",
        "pd.set_option('display.max_columns', None)\n",
        "pd.set_option('display.max_colwidth', None)\n",
        "\n",
        "#converte as 30 amostras filtradas pelo vep-gc.sh em amostras .csv --> output salvo na pasta /content/lmabrasil-hg38/csv_filtrados\n",
        "!mkdir /content/lmabrasil-hg38/csv_filtrados\n",
        "for i in SAMPLES:\n",
        "  df = pd.read_csv(f'/content/lmabrasil-hg38/vep_output/liftOver_{i}_hg19ToHg38.vep.filter.tsv',sep='\\t',index_col=False)\n",
        "  df.to_csv(f'/content/lmabrasil-hg38/csv_filtrados/{i}_filtrado.csv', index=False)"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "3m-5BHEbQ775"
      },
      "source": [
        "## **6. TABELA BRUTA COM VARIANTES DO PAINEL DE GENES (RESULTADO)**\n",
        "\n",
        "Resultado da filtragem das 30 amostras para mutações de prognóstico ruim para mielofibrose - FALTANDO FILTRAGEM DE SCORE\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 603
        },
        "id": "m2Y0OJxAOAVM",
        "outputId": "b6b81bb6-aa53-440d-d39c-a8dacce8b914"
      },
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "    CHROM       POS                                                    REF  \\\n",
              "0   chr19  12943750  AGCAGAGGCTTAAGGAGGAGGAAGAAGACAAGAAACGCAAAGAGGAGGAGGAG   \n",
              "1   chr20  32434638                                                      A   \n",
              "2   chr20  32434483                                                      T   \n",
              "3   chr21  43104346                                                      G   \n",
              "4   chr20  32434789                                                      C   \n",
              "5   chr21  43094667                                                      T   \n",
              "6   chr19  12943750  AGCAGAGGCTTAAGGAGGAGGAAGAAGACAAGAAACGCAAAGAGGAGGAGGAG   \n",
              "7   chr19  12943813                                                      A   \n",
              "8   chr19  12943750  AGCAGAGGCTTAAGGAGGAGGAAGAAGACAAGAAACGCAAAGAGGAGGAGGAG   \n",
              "9   chr19  12943750  AGCAGAGGCTTAAGGAGGAGGAAGAAGACAAGAAACGCAAAGAGGAGGAGGAG   \n",
              "10  chr20  32434599                               TCACCACTGCCATAGAGAGGCGGC   \n",
              "11  chr19  12943813                                                      A   \n",
              "12  chr17  76736877                                                      G   \n",
              "13  chr17  76736877                                                      G   \n",
              "14  chr19  12943750  AGCAGAGGCTTAAGGAGGAGGAAGAAGACAAGAAACGCAAAGAGGAGGAGGAG   \n",
              "15  chr19  12943750  AGCAGAGGCTTAAGGAGGAGGAAGAAGACAAGAAACGCAAAGAGGAGGAGGAG   \n",
              "16  chr20  32434599                               TCACCACTGCCATAGAGAGGCGGC   \n",
              "\n",
              "       ALT                 Location SYMBOL                     Consequence  \\\n",
              "0        A  chr19:12943751-12943802   CALR              frameshift_variant   \n",
              "1       AG  chr20:32434638-32434639  ASXL1              frameshift_variant   \n",
              "2       TA  chr20:32434483-32434484  ASXL1  stop_gained&frameshift_variant   \n",
              "3        A           chr21:43104346  U2AF1                missense_variant   \n",
              "4        T           chr20:32434789  ASXL1                     stop_gained   \n",
              "5        G           chr21:43094667  U2AF1                missense_variant   \n",
              "6        A  chr19:12943751-12943802   CALR              frameshift_variant   \n",
              "7   ATTGTC  chr19:12943813-12943814   CALR              frameshift_variant   \n",
              "8        A  chr19:12943751-12943802   CALR              frameshift_variant   \n",
              "9        A  chr19:12943751-12943802   CALR              frameshift_variant   \n",
              "10       T  chr20:32434600-32434622  ASXL1              frameshift_variant   \n",
              "11  ATTGTC  chr19:12943813-12943814   CALR              frameshift_variant   \n",
              "12       T           chr17:76736877  SRSF2                missense_variant   \n",
              "13       A           chr17:76736877  SRSF2                missense_variant   \n",
              "14       A  chr19:12943751-12943802   CALR              frameshift_variant   \n",
              "15       A  chr19:12943751-12943802   CALR              frameshift_variant   \n",
              "16       T  chr20:32434600-32434622  ASXL1              frameshift_variant   \n",
              "\n",
              "           Feature         MANE_SELECT         BIOTYPE  \\\n",
              "0      NM_004343.4  ENST00000316448.10  protein_coding   \n",
              "1      NM_015338.6  ENST00000375687.10  protein_coding   \n",
              "2      NM_015338.6  ENST00000375687.10  protein_coding   \n",
              "3      NM_006758.3   ENST00000291552.9  protein_coding   \n",
              "4      NM_015338.6  ENST00000375687.10  protein_coding   \n",
              "5      NM_006758.3   ENST00000291552.9  protein_coding   \n",
              "6      NM_004343.4  ENST00000316448.10  protein_coding   \n",
              "7      NM_004343.4  ENST00000316448.10  protein_coding   \n",
              "8      NM_004343.4  ENST00000316448.10  protein_coding   \n",
              "9      NM_004343.4  ENST00000316448.10  protein_coding   \n",
              "10     NM_015338.6  ENST00000375687.10  protein_coding   \n",
              "11     NM_004343.4  ENST00000316448.10  protein_coding   \n",
              "12  NM_001195427.2  ENST00000359995.10  protein_coding   \n",
              "13  NM_001195427.2  ENST00000359995.10  protein_coding   \n",
              "14     NM_004343.4  ENST00000316448.10  protein_coding   \n",
              "15     NM_004343.4  ENST00000316448.10  protein_coding   \n",
              "16     NM_015338.6  ENST00000375687.10  protein_coding   \n",
              "\n",
              "                              HGVSc                           HGVSp   EXON  \\\n",
              "0        NM_004343.4:c.1099_1150del  NP_004334.1:p.Leu367ThrfsTer46    9/9   \n",
              "1             NM_015338.6:c.1934dup  NP_056153.2:p.Gly646TrpfsTer12  13/13   \n",
              "2             NM_015338.6:c.1772dup         NP_056153.2:p.Tyr591Ter  13/13   \n",
              "3              NM_006758.3:c.101C>T          NP_006749.1:p.Ser34Phe    2/8   \n",
              "4             NM_015338.6:c.2077C>T         NP_056153.2:p.Arg693Ter  13/13   \n",
              "5              NM_006758.3:c.470A>C         NP_006749.1:p.Gln157Pro    6/8   \n",
              "6        NM_004343.4:c.1099_1150del  NP_004334.1:p.Leu367ThrfsTer46    9/9   \n",
              "7   NM_004343.4:c.1154_1155insTTGTC  NP_004334.1:p.Lys385AsnfsTer47    9/9   \n",
              "8        NM_004343.4:c.1099_1150del  NP_004334.1:p.Leu367ThrfsTer46    9/9   \n",
              "9        NM_004343.4:c.1099_1150del  NP_004334.1:p.Leu367ThrfsTer46    9/9   \n",
              "10       NM_015338.6:c.1900_1922del  NP_056153.2:p.Glu635ArgfsTer15  13/13   \n",
              "11  NM_004343.4:c.1154_1155insTTGTC  NP_004334.1:p.Lys385AsnfsTer47    9/9   \n",
              "12          NM_001195427.2:c.284C>A       NP_001182356.1:p.Pro95His    1/3   \n",
              "13          NM_001195427.2:c.284C>T       NP_001182356.1:p.Pro95Leu    1/3   \n",
              "14       NM_004343.4:c.1099_1150del  NP_004334.1:p.Leu367ThrfsTer46    9/9   \n",
              "15       NM_004343.4:c.1099_1150del  NP_004334.1:p.Leu367ThrfsTer46    9/9   \n",
              "16       NM_015338.6:c.1900_1922del  NP_056153.2:p.Glu635ArgfsTer15  13/13   \n",
              "\n",
              "   INTRON VARIANT_CLASS                              SIFT  \\\n",
              "0       .      deletion                                 .   \n",
              "1       .     insertion                                 .   \n",
              "2       .     insertion                                 .   \n",
              "3       .           SNV  deleterious_low_confidence(0.01)   \n",
              "4       .           SNV                                 .   \n",
              "5       .           SNV     deleterious_low_confidence(0)   \n",
              "6       .      deletion                                 .   \n",
              "7       .     insertion                                 .   \n",
              "8       .      deletion                                 .   \n",
              "9       .      deletion                                 .   \n",
              "10      .      deletion                                 .   \n",
              "11      .     insertion                                 .   \n",
              "12      .           SNV                    deleterious(0)   \n",
              "13      .           SNV                   tolerated(0.17)   \n",
              "14      .      deletion                                 .   \n",
              "15      .      deletion                                 .   \n",
              "16      .      deletion                                 .   \n",
              "\n",
              "                    PolyPhen  gnomADg_AF    MAX_AF    IMPACT  \\\n",
              "0                          .    0.000020  0.000066      HIGH   \n",
              "1                          .    0.000644  0.001739      HIGH   \n",
              "2                          .    0.000033  0.000094      HIGH   \n",
              "3   probably_damaging(0.998)    0.000000  0.000000  MODERATE   \n",
              "4                          .    0.000020  0.000062      HIGH   \n",
              "5   possibly_damaging(0.541)    0.000073  0.000115  MODERATE   \n",
              "6                          .    0.000020  0.000066      HIGH   \n",
              "7                          .    0.000020  0.000094      HIGH   \n",
              "8                          .    0.000020  0.000066      HIGH   \n",
              "9                          .    0.000020  0.000066      HIGH   \n",
              "10                         .    0.000113  0.000236      HIGH   \n",
              "11                         .    0.000020  0.000094      HIGH   \n",
              "12                         .    0.000118  0.000236  MODERATE   \n",
              "13                         .    0.000092  0.000207  MODERATE   \n",
              "14                         .    0.000020  0.000066      HIGH   \n",
              "15                         .    0.000020  0.000066      HIGH   \n",
              "16                         .    0.000113  0.000236      HIGH   \n",
              "\n",
              "                          CLIN_SIG    SOMATIC  \\\n",
              "0                       pathogenic          .   \n",
              "1     likely_pathogenic&pathogenic        0&1   \n",
              "2                                .      0&1&1   \n",
              "3                likely_pathogenic      0&1&1   \n",
              "4     pathogenic&likely_pathogenic        0&1   \n",
              "5   not_provided&likely_pathogenic      0&1&1   \n",
              "6                       pathogenic          .   \n",
              "7                       pathogenic      0&1&1   \n",
              "8                       pathogenic          .   \n",
              "9                       pathogenic          .   \n",
              "10    likely_pathogenic&pathogenic          .   \n",
              "11                      pathogenic      0&1&1   \n",
              "12                               .  0&0&1&1&1   \n",
              "13                               .  0&0&1&1&1   \n",
              "14                      pathogenic          .   \n",
              "15                      pathogenic          .   \n",
              "16    likely_pathogenic&pathogenic          .   \n",
              "\n",
              "                                              Existing_variation  \\\n",
              "0                                                   rs1555760738   \n",
              "1                                       rs750318549&COSV60102510   \n",
              "2                          rs762036456&COSV60104310&COSV60124684   \n",
              "3                          rs371769427&COSV52341059&COSV52341472   \n",
              "4                                       rs373221034&COSV60102332   \n",
              "5                          rs371246226&COSV52341120&COSV52341147   \n",
              "6                                                   rs1555760738   \n",
              "7                          rs765476509&COSV57116551&COSV57118194   \n",
              "8                                                   rs1555760738   \n",
              "9                                                   rs1555760738   \n",
              "10                                                   rs766433101   \n",
              "11                         rs765476509&COSV57116551&COSV57118194   \n",
              "12  rs751713049&CM1511359&COSV57969809&COSV57969816&COSV57969830   \n",
              "13  rs751713049&CM1511359&COSV57969809&COSV57969816&COSV57969830   \n",
              "14                                                  rs1555760738   \n",
              "15                                                  rs1555760738   \n",
              "16                                                   rs766433101   \n",
              "\n",
              "                      FILTER TumorID   GT   DP       AD     AF NormalID  NGT  \\\n",
              "0                   map_qual   WP078  0/1   91    59,32  0.406    WP079  0/0   \n",
              "1                       PASS   WP078  0/1   53    25,28  0.536    WP079  0/0   \n",
              "2            normal_artifact   WP093  0/1  312  161,151  0.481    WP094  0/0   \n",
              "3                       PASS   WP216  0/1  106    77,29  0.284    WP217  0/0   \n",
              "4                       PASS   WP306  0/1  130    77,53  0.434    WP307  0/0   \n",
              "5                       PASS   WP306  0/1   82    49,33  0.429    WP307  0/0   \n",
              "6                   map_qual   WP297  0/1   66    47,19  0.387    WP298  0/0   \n",
              "7                   germline   WP276  0/1   30     22,8  0.256      NaN  NaN   \n",
              "8                   map_qual   WP188  0/1   61    46,15  0.288    WP189  0/0   \n",
              "9   map_qual;normal_artifact   WP160  0/1   45    35,10  0.261    WP161  0/0   \n",
              "10                      PASS   WP056  0/1   87    48,39  0.491    WP057  0/0   \n",
              "11  germline;normal_artifact   WP274  0/1   36    22,14  0.417    WP275  0/0   \n",
              "12                      PASS   WP212  0/1  229  120,109  0.478    WP213  0/0   \n",
              "13                      PASS   WP060  0/1  247  119,128  0.525    WP061  0/0   \n",
              "14                  map_qual   WP140  0/1   54    35,19  0.399    WP141  0/0   \n",
              "15                  map_qual   WP270  0/1   40    26,14  0.414    WP271  0/0   \n",
              "16                      PASS   WP270  0/1  105    68,37  0.395    WP271  0/0   \n",
              "\n",
              "      NDP    NAD    NAF  \n",
              "0    46.0   46,0  0.028  \n",
              "1    31.0   30,1  0.029  \n",
              "2    91.0   86,5  0.073  \n",
              "3    49.0   48,1  0.044  \n",
              "4   132.0  131,1  0.019  \n",
              "5    60.0   59,1  0.037  \n",
              "6    27.0   27,0  0.045  \n",
              "7     NaN    NaN    NaN  \n",
              "8    29.0   29,0  0.041  \n",
              "9    41.0   40,1  0.059  \n",
              "10   40.0   40,0  0.024  \n",
              "11   25.0   24,1  0.080  \n",
              "12  103.0  102,1  0.023  \n",
              "13  101.0   99,2  0.034  \n",
              "14   25.0   25,0  0.044  \n",
              "15   38.0   38,0  0.031  \n",
              "16   73.0   73,0  0.015  "
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-e2189fab-a617-4610-ae77-3ad761f9c62a\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>CHROM</th>\n",
              "      <th>POS</th>\n",
              "      <th>REF</th>\n",
              "      <th>ALT</th>\n",
              "      <th>Location</th>\n",
              "      <th>SYMBOL</th>\n",
              "      <th>Consequence</th>\n",
              "      <th>Feature</th>\n",
              "      <th>MANE_SELECT</th>\n",
              "      <th>BIOTYPE</th>\n",
              "      <th>HGVSc</th>\n",
              "      <th>HGVSp</th>\n",
              "      <th>EXON</th>\n",
              "      <th>INTRON</th>\n",
              "      <th>VARIANT_CLASS</th>\n",
              "      <th>SIFT</th>\n",
              "      <th>PolyPhen</th>\n",
              "      <th>gnomADg_AF</th>\n",
              "      <th>MAX_AF</th>\n",
              "      <th>IMPACT</th>\n",
              "      <th>CLIN_SIG</th>\n",
              "      <th>SOMATIC</th>\n",
              "      <th>Existing_variation</th>\n",
              "      <th>FILTER</th>\n",
              "      <th>TumorID</th>\n",
              "      <th>GT</th>\n",
              "      <th>DP</th>\n",
              "      <th>AD</th>\n",
              "      <th>AF</th>\n",
              "      <th>NormalID</th>\n",
              "      <th>NGT</th>\n",
              "      <th>NDP</th>\n",
              "      <th>NAD</th>\n",
              "      <th>NAF</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>0</th>\n",
              "      <td>chr19</td>\n",
              "      <td>12943750</td>\n",
              "      <td>AGCAGAGGCTTAAGGAGGAGGAAGAAGACAAGAAACGCAAAGAGGAGGAGGAG</td>\n",
              "      <td>A</td>\n",
              "      <td>chr19:12943751-12943802</td>\n",
              "      <td>CALR</td>\n",
              "      <td>frameshift_variant</td>\n",
              "      <td>NM_004343.4</td>\n",
              "      <td>ENST00000316448.10</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>NM_004343.4:c.1099_1150del</td>\n",
              "      <td>NP_004334.1:p.Leu367ThrfsTer46</td>\n",
              "      <td>9/9</td>\n",
              "      <td>.</td>\n",
              "      <td>deletion</td>\n",
              "      <td>.</td>\n",
              "      <td>.</td>\n",
              "      <td>0.000020</td>\n",
              "      <td>0.000066</td>\n",
              "      <td>HIGH</td>\n",
              "      <td>pathogenic</td>\n",
              "      <td>.</td>\n",
              "      <td>rs1555760738</td>\n",
              "      <td>map_qual</td>\n",
              "      <td>WP078</td>\n",
              "      <td>0/1</td>\n",
              "      <td>91</td>\n",
              "      <td>59,32</td>\n",
              "      <td>0.406</td>\n",
              "      <td>WP079</td>\n",
              "      <td>0/0</td>\n",
              "      <td>46.0</td>\n",
              "      <td>46,0</td>\n",
              "      <td>0.028</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>1</th>\n",
              "      <td>chr20</td>\n",
              "      <td>32434638</td>\n",
              "      <td>A</td>\n",
              "      <td>AG</td>\n",
              "      <td>chr20:32434638-32434639</td>\n",
              "      <td>ASXL1</td>\n",
              "      <td>frameshift_variant</td>\n",
              "      <td>NM_015338.6</td>\n",
              "      <td>ENST00000375687.10</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>NM_015338.6:c.1934dup</td>\n",
              "      <td>NP_056153.2:p.Gly646TrpfsTer12</td>\n",
              "      <td>13/13</td>\n",
              "      <td>.</td>\n",
              "      <td>insertion</td>\n",
              "      <td>.</td>\n",
              "      <td>.</td>\n",
              "      <td>0.000644</td>\n",
              "      <td>0.001739</td>\n",
              "      <td>HIGH</td>\n",
              "      <td>likely_pathogenic&amp;pathogenic</td>\n",
              "      <td>0&amp;1</td>\n",
              "      <td>rs750318549&amp;COSV60102510</td>\n",
              "      <td>PASS</td>\n",
              "      <td>WP078</td>\n",
              "      <td>0/1</td>\n",
              "      <td>53</td>\n",
              "      <td>25,28</td>\n",
              "      <td>0.536</td>\n",
              "      <td>WP079</td>\n",
              "      <td>0/0</td>\n",
              "      <td>31.0</td>\n",
              "      <td>30,1</td>\n",
              "      <td>0.029</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>2</th>\n",
              "      <td>chr20</td>\n",
              "      <td>32434483</td>\n",
              "      <td>T</td>\n",
              "      <td>TA</td>\n",
              "      <td>chr20:32434483-32434484</td>\n",
              "      <td>ASXL1</td>\n",
              "      <td>stop_gained&amp;frameshift_variant</td>\n",
              "      <td>NM_015338.6</td>\n",
              "      <td>ENST00000375687.10</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>NM_015338.6:c.1772dup</td>\n",
              "      <td>NP_056153.2:p.Tyr591Ter</td>\n",
              "      <td>13/13</td>\n",
              "      <td>.</td>\n",
              "      <td>insertion</td>\n",
              "      <td>.</td>\n",
              "      <td>.</td>\n",
              "      <td>0.000033</td>\n",
              "      <td>0.000094</td>\n",
              "      <td>HIGH</td>\n",
              "      <td>.</td>\n",
              "      <td>0&amp;1&amp;1</td>\n",
              "      <td>rs762036456&amp;COSV60104310&amp;COSV60124684</td>\n",
              "      <td>normal_artifact</td>\n",
              "      <td>WP093</td>\n",
              "      <td>0/1</td>\n",
              "      <td>312</td>\n",
              "      <td>161,151</td>\n",
              "      <td>0.481</td>\n",
              "      <td>WP094</td>\n",
              "      <td>0/0</td>\n",
              "      <td>91.0</td>\n",
              "      <td>86,5</td>\n",
              "      <td>0.073</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>3</th>\n",
              "      <td>chr21</td>\n",
              "      <td>43104346</td>\n",
              "      <td>G</td>\n",
              "      <td>A</td>\n",
              "      <td>chr21:43104346</td>\n",
              "      <td>U2AF1</td>\n",
              "      <td>missense_variant</td>\n",
              "      <td>NM_006758.3</td>\n",
              "      <td>ENST00000291552.9</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>NM_006758.3:c.101C&gt;T</td>\n",
              "      <td>NP_006749.1:p.Ser34Phe</td>\n",
              "      <td>2/8</td>\n",
              "      <td>.</td>\n",
              "      <td>SNV</td>\n",
              "      <td>deleterious_low_confidence(0.01)</td>\n",
              "      <td>probably_damaging(0.998)</td>\n",
              "      <td>0.000000</td>\n",
              "      <td>0.000000</td>\n",
              "      <td>MODERATE</td>\n",
              "      <td>likely_pathogenic</td>\n",
              "      <td>0&amp;1&amp;1</td>\n",
              "      <td>rs371769427&amp;COSV52341059&amp;COSV52341472</td>\n",
              "      <td>PASS</td>\n",
              "      <td>WP216</td>\n",
              "      <td>0/1</td>\n",
              "      <td>106</td>\n",
              "      <td>77,29</td>\n",
              "      <td>0.284</td>\n",
              "      <td>WP217</td>\n",
              "      <td>0/0</td>\n",
              "      <td>49.0</td>\n",
              "      <td>48,1</td>\n",
              "      <td>0.044</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>4</th>\n",
              "      <td>chr20</td>\n",
              "      <td>32434789</td>\n",
              "      <td>C</td>\n",
              "      <td>T</td>\n",
              "      <td>chr20:32434789</td>\n",
              "      <td>ASXL1</td>\n",
              "      <td>stop_gained</td>\n",
              "      <td>NM_015338.6</td>\n",
              "      <td>ENST00000375687.10</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>NM_015338.6:c.2077C&gt;T</td>\n",
              "      <td>NP_056153.2:p.Arg693Ter</td>\n",
              "      <td>13/13</td>\n",
              "      <td>.</td>\n",
              "      <td>SNV</td>\n",
              "      <td>.</td>\n",
              "      <td>.</td>\n",
              "      <td>0.000020</td>\n",
              "      <td>0.000062</td>\n",
              "      <td>HIGH</td>\n",
              "      <td>pathogenic&amp;likely_pathogenic</td>\n",
              "      <td>0&amp;1</td>\n",
              "      <td>rs373221034&amp;COSV60102332</td>\n",
              "      <td>PASS</td>\n",
              "      <td>WP306</td>\n",
              "      <td>0/1</td>\n",
              "      <td>130</td>\n",
              "      <td>77,53</td>\n",
              "      <td>0.434</td>\n",
              "      <td>WP307</td>\n",
              "      <td>0/0</td>\n",
              "      <td>132.0</td>\n",
              "      <td>131,1</td>\n",
              "      <td>0.019</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5</th>\n",
              "      <td>chr21</td>\n",
              "      <td>43094667</td>\n",
              "      <td>T</td>\n",
              "      <td>G</td>\n",
              "      <td>chr21:43094667</td>\n",
              "      <td>U2AF1</td>\n",
              "      <td>missense_variant</td>\n",
              "      <td>NM_006758.3</td>\n",
              "      <td>ENST00000291552.9</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>NM_006758.3:c.470A&gt;C</td>\n",
              "      <td>NP_006749.1:p.Gln157Pro</td>\n",
              "      <td>6/8</td>\n",
              "      <td>.</td>\n",
              "      <td>SNV</td>\n",
              "      <td>deleterious_low_confidence(0)</td>\n",
              "      <td>possibly_damaging(0.541)</td>\n",
              "      <td>0.000073</td>\n",
              "      <td>0.000115</td>\n",
              "      <td>MODERATE</td>\n",
              "      <td>not_provided&amp;likely_pathogenic</td>\n",
              "      <td>0&amp;1&amp;1</td>\n",
              "      <td>rs371246226&amp;COSV52341120&amp;COSV52341147</td>\n",
              "      <td>PASS</td>\n",
              "      <td>WP306</td>\n",
              "      <td>0/1</td>\n",
              "      <td>82</td>\n",
              "      <td>49,33</td>\n",
              "      <td>0.429</td>\n",
              "      <td>WP307</td>\n",
              "      <td>0/0</td>\n",
              "      <td>60.0</td>\n",
              "      <td>59,1</td>\n",
              "      <td>0.037</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>6</th>\n",
              "      <td>chr19</td>\n",
              "      <td>12943750</td>\n",
              "      <td>AGCAGAGGCTTAAGGAGGAGGAAGAAGACAAGAAACGCAAAGAGGAGGAGGAG</td>\n",
              "      <td>A</td>\n",
              "      <td>chr19:12943751-12943802</td>\n",
              "      <td>CALR</td>\n",
              "      <td>frameshift_variant</td>\n",
              "      <td>NM_004343.4</td>\n",
              "      <td>ENST00000316448.10</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>NM_004343.4:c.1099_1150del</td>\n",
              "      <td>NP_004334.1:p.Leu367ThrfsTer46</td>\n",
              "      <td>9/9</td>\n",
              "      <td>.</td>\n",
              "      <td>deletion</td>\n",
              "      <td>.</td>\n",
              "      <td>.</td>\n",
              "      <td>0.000020</td>\n",
              "      <td>0.000066</td>\n",
              "      <td>HIGH</td>\n",
              "      <td>pathogenic</td>\n",
              "      <td>.</td>\n",
              "      <td>rs1555760738</td>\n",
              "      <td>map_qual</td>\n",
              "      <td>WP297</td>\n",
              "      <td>0/1</td>\n",
              "      <td>66</td>\n",
              "      <td>47,19</td>\n",
              "      <td>0.387</td>\n",
              "      <td>WP298</td>\n",
              "      <td>0/0</td>\n",
              "      <td>27.0</td>\n",
              "      <td>27,0</td>\n",
              "      <td>0.045</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>7</th>\n",
              "      <td>chr19</td>\n",
              "      <td>12943813</td>\n",
              "      <td>A</td>\n",
              "      <td>ATTGTC</td>\n",
              "      <td>chr19:12943813-12943814</td>\n",
              "      <td>CALR</td>\n",
              "      <td>frameshift_variant</td>\n",
              "      <td>NM_004343.4</td>\n",
              "      <td>ENST00000316448.10</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>NM_004343.4:c.1154_1155insTTGTC</td>\n",
              "      <td>NP_004334.1:p.Lys385AsnfsTer47</td>\n",
              "      <td>9/9</td>\n",
              "      <td>.</td>\n",
              "      <td>insertion</td>\n",
              "      <td>.</td>\n",
              "      <td>.</td>\n",
              "      <td>0.000020</td>\n",
              "      <td>0.000094</td>\n",
              "      <td>HIGH</td>\n",
              "      <td>pathogenic</td>\n",
              "      <td>0&amp;1&amp;1</td>\n",
              "      <td>rs765476509&amp;COSV57116551&amp;COSV57118194</td>\n",
              "      <td>germline</td>\n",
              "      <td>WP276</td>\n",
              "      <td>0/1</td>\n",
              "      <td>30</td>\n",
              "      <td>22,8</td>\n",
              "      <td>0.256</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "      <td>NaN</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>8</th>\n",
              "      <td>chr19</td>\n",
              "      <td>12943750</td>\n",
              "      <td>AGCAGAGGCTTAAGGAGGAGGAAGAAGACAAGAAACGCAAAGAGGAGGAGGAG</td>\n",
              "      <td>A</td>\n",
              "      <td>chr19:12943751-12943802</td>\n",
              "      <td>CALR</td>\n",
              "      <td>frameshift_variant</td>\n",
              "      <td>NM_004343.4</td>\n",
              "      <td>ENST00000316448.10</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>NM_004343.4:c.1099_1150del</td>\n",
              "      <td>NP_004334.1:p.Leu367ThrfsTer46</td>\n",
              "      <td>9/9</td>\n",
              "      <td>.</td>\n",
              "      <td>deletion</td>\n",
              "      <td>.</td>\n",
              "      <td>.</td>\n",
              "      <td>0.000020</td>\n",
              "      <td>0.000066</td>\n",
              "      <td>HIGH</td>\n",
              "      <td>pathogenic</td>\n",
              "      <td>.</td>\n",
              "      <td>rs1555760738</td>\n",
              "      <td>map_qual</td>\n",
              "      <td>WP188</td>\n",
              "      <td>0/1</td>\n",
              "      <td>61</td>\n",
              "      <td>46,15</td>\n",
              "      <td>0.288</td>\n",
              "      <td>WP189</td>\n",
              "      <td>0/0</td>\n",
              "      <td>29.0</td>\n",
              "      <td>29,0</td>\n",
              "      <td>0.041</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>9</th>\n",
              "      <td>chr19</td>\n",
              "      <td>12943750</td>\n",
              "      <td>AGCAGAGGCTTAAGGAGGAGGAAGAAGACAAGAAACGCAAAGAGGAGGAGGAG</td>\n",
              "      <td>A</td>\n",
              "      <td>chr19:12943751-12943802</td>\n",
              "      <td>CALR</td>\n",
              "      <td>frameshift_variant</td>\n",
              "      <td>NM_004343.4</td>\n",
              "      <td>ENST00000316448.10</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>NM_004343.4:c.1099_1150del</td>\n",
              "      <td>NP_004334.1:p.Leu367ThrfsTer46</td>\n",
              "      <td>9/9</td>\n",
              "      <td>.</td>\n",
              "      <td>deletion</td>\n",
              "      <td>.</td>\n",
              "      <td>.</td>\n",
              "      <td>0.000020</td>\n",
              "      <td>0.000066</td>\n",
              "      <td>HIGH</td>\n",
              "      <td>pathogenic</td>\n",
              "      <td>.</td>\n",
              "      <td>rs1555760738</td>\n",
              "      <td>map_qual;normal_artifact</td>\n",
              "      <td>WP160</td>\n",
              "      <td>0/1</td>\n",
              "      <td>45</td>\n",
              "      <td>35,10</td>\n",
              "      <td>0.261</td>\n",
              "      <td>WP161</td>\n",
              "      <td>0/0</td>\n",
              "      <td>41.0</td>\n",
              "      <td>40,1</td>\n",
              "      <td>0.059</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>10</th>\n",
              "      <td>chr20</td>\n",
              "      <td>32434599</td>\n",
              "      <td>TCACCACTGCCATAGAGAGGCGGC</td>\n",
              "      <td>T</td>\n",
              "      <td>chr20:32434600-32434622</td>\n",
              "      <td>ASXL1</td>\n",
              "      <td>frameshift_variant</td>\n",
              "      <td>NM_015338.6</td>\n",
              "      <td>ENST00000375687.10</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>NM_015338.6:c.1900_1922del</td>\n",
              "      <td>NP_056153.2:p.Glu635ArgfsTer15</td>\n",
              "      <td>13/13</td>\n",
              "      <td>.</td>\n",
              "      <td>deletion</td>\n",
              "      <td>.</td>\n",
              "      <td>.</td>\n",
              "      <td>0.000113</td>\n",
              "      <td>0.000236</td>\n",
              "      <td>HIGH</td>\n",
              "      <td>likely_pathogenic&amp;pathogenic</td>\n",
              "      <td>.</td>\n",
              "      <td>rs766433101</td>\n",
              "      <td>PASS</td>\n",
              "      <td>WP056</td>\n",
              "      <td>0/1</td>\n",
              "      <td>87</td>\n",
              "      <td>48,39</td>\n",
              "      <td>0.491</td>\n",
              "      <td>WP057</td>\n",
              "      <td>0/0</td>\n",
              "      <td>40.0</td>\n",
              "      <td>40,0</td>\n",
              "      <td>0.024</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>11</th>\n",
              "      <td>chr19</td>\n",
              "      <td>12943813</td>\n",
              "      <td>A</td>\n",
              "      <td>ATTGTC</td>\n",
              "      <td>chr19:12943813-12943814</td>\n",
              "      <td>CALR</td>\n",
              "      <td>frameshift_variant</td>\n",
              "      <td>NM_004343.4</td>\n",
              "      <td>ENST00000316448.10</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>NM_004343.4:c.1154_1155insTTGTC</td>\n",
              "      <td>NP_004334.1:p.Lys385AsnfsTer47</td>\n",
              "      <td>9/9</td>\n",
              "      <td>.</td>\n",
              "      <td>insertion</td>\n",
              "      <td>.</td>\n",
              "      <td>.</td>\n",
              "      <td>0.000020</td>\n",
              "      <td>0.000094</td>\n",
              "      <td>HIGH</td>\n",
              "      <td>pathogenic</td>\n",
              "      <td>0&amp;1&amp;1</td>\n",
              "      <td>rs765476509&amp;COSV57116551&amp;COSV57118194</td>\n",
              "      <td>germline;normal_artifact</td>\n",
              "      <td>WP274</td>\n",
              "      <td>0/1</td>\n",
              "      <td>36</td>\n",
              "      <td>22,14</td>\n",
              "      <td>0.417</td>\n",
              "      <td>WP275</td>\n",
              "      <td>0/0</td>\n",
              "      <td>25.0</td>\n",
              "      <td>24,1</td>\n",
              "      <td>0.080</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>12</th>\n",
              "      <td>chr17</td>\n",
              "      <td>76736877</td>\n",
              "      <td>G</td>\n",
              "      <td>T</td>\n",
              "      <td>chr17:76736877</td>\n",
              "      <td>SRSF2</td>\n",
              "      <td>missense_variant</td>\n",
              "      <td>NM_001195427.2</td>\n",
              "      <td>ENST00000359995.10</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>NM_001195427.2:c.284C&gt;A</td>\n",
              "      <td>NP_001182356.1:p.Pro95His</td>\n",
              "      <td>1/3</td>\n",
              "      <td>.</td>\n",
              "      <td>SNV</td>\n",
              "      <td>deleterious(0)</td>\n",
              "      <td>.</td>\n",
              "      <td>0.000118</td>\n",
              "      <td>0.000236</td>\n",
              "      <td>MODERATE</td>\n",
              "      <td>.</td>\n",
              "      <td>0&amp;0&amp;1&amp;1&amp;1</td>\n",
              "      <td>rs751713049&amp;CM1511359&amp;COSV57969809&amp;COSV57969816&amp;COSV57969830</td>\n",
              "      <td>PASS</td>\n",
              "      <td>WP212</td>\n",
              "      <td>0/1</td>\n",
              "      <td>229</td>\n",
              "      <td>120,109</td>\n",
              "      <td>0.478</td>\n",
              "      <td>WP213</td>\n",
              "      <td>0/0</td>\n",
              "      <td>103.0</td>\n",
              "      <td>102,1</td>\n",
              "      <td>0.023</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>13</th>\n",
              "      <td>chr17</td>\n",
              "      <td>76736877</td>\n",
              "      <td>G</td>\n",
              "      <td>A</td>\n",
              "      <td>chr17:76736877</td>\n",
              "      <td>SRSF2</td>\n",
              "      <td>missense_variant</td>\n",
              "      <td>NM_001195427.2</td>\n",
              "      <td>ENST00000359995.10</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>NM_001195427.2:c.284C&gt;T</td>\n",
              "      <td>NP_001182356.1:p.Pro95Leu</td>\n",
              "      <td>1/3</td>\n",
              "      <td>.</td>\n",
              "      <td>SNV</td>\n",
              "      <td>tolerated(0.17)</td>\n",
              "      <td>.</td>\n",
              "      <td>0.000092</td>\n",
              "      <td>0.000207</td>\n",
              "      <td>MODERATE</td>\n",
              "      <td>.</td>\n",
              "      <td>0&amp;0&amp;1&amp;1&amp;1</td>\n",
              "      <td>rs751713049&amp;CM1511359&amp;COSV57969809&amp;COSV57969816&amp;COSV57969830</td>\n",
              "      <td>PASS</td>\n",
              "      <td>WP060</td>\n",
              "      <td>0/1</td>\n",
              "      <td>247</td>\n",
              "      <td>119,128</td>\n",
              "      <td>0.525</td>\n",
              "      <td>WP061</td>\n",
              "      <td>0/0</td>\n",
              "      <td>101.0</td>\n",
              "      <td>99,2</td>\n",
              "      <td>0.034</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>14</th>\n",
              "      <td>chr19</td>\n",
              "      <td>12943750</td>\n",
              "      <td>AGCAGAGGCTTAAGGAGGAGGAAGAAGACAAGAAACGCAAAGAGGAGGAGGAG</td>\n",
              "      <td>A</td>\n",
              "      <td>chr19:12943751-12943802</td>\n",
              "      <td>CALR</td>\n",
              "      <td>frameshift_variant</td>\n",
              "      <td>NM_004343.4</td>\n",
              "      <td>ENST00000316448.10</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>NM_004343.4:c.1099_1150del</td>\n",
              "      <td>NP_004334.1:p.Leu367ThrfsTer46</td>\n",
              "      <td>9/9</td>\n",
              "      <td>.</td>\n",
              "      <td>deletion</td>\n",
              "      <td>.</td>\n",
              "      <td>.</td>\n",
              "      <td>0.000020</td>\n",
              "      <td>0.000066</td>\n",
              "      <td>HIGH</td>\n",
              "      <td>pathogenic</td>\n",
              "      <td>.</td>\n",
              "      <td>rs1555760738</td>\n",
              "      <td>map_qual</td>\n",
              "      <td>WP140</td>\n",
              "      <td>0/1</td>\n",
              "      <td>54</td>\n",
              "      <td>35,19</td>\n",
              "      <td>0.399</td>\n",
              "      <td>WP141</td>\n",
              "      <td>0/0</td>\n",
              "      <td>25.0</td>\n",
              "      <td>25,0</td>\n",
              "      <td>0.044</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>15</th>\n",
              "      <td>chr19</td>\n",
              "      <td>12943750</td>\n",
              "      <td>AGCAGAGGCTTAAGGAGGAGGAAGAAGACAAGAAACGCAAAGAGGAGGAGGAG</td>\n",
              "      <td>A</td>\n",
              "      <td>chr19:12943751-12943802</td>\n",
              "      <td>CALR</td>\n",
              "      <td>frameshift_variant</td>\n",
              "      <td>NM_004343.4</td>\n",
              "      <td>ENST00000316448.10</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>NM_004343.4:c.1099_1150del</td>\n",
              "      <td>NP_004334.1:p.Leu367ThrfsTer46</td>\n",
              "      <td>9/9</td>\n",
              "      <td>.</td>\n",
              "      <td>deletion</td>\n",
              "      <td>.</td>\n",
              "      <td>.</td>\n",
              "      <td>0.000020</td>\n",
              "      <td>0.000066</td>\n",
              "      <td>HIGH</td>\n",
              "      <td>pathogenic</td>\n",
              "      <td>.</td>\n",
              "      <td>rs1555760738</td>\n",
              "      <td>map_qual</td>\n",
              "      <td>WP270</td>\n",
              "      <td>0/1</td>\n",
              "      <td>40</td>\n",
              "      <td>26,14</td>\n",
              "      <td>0.414</td>\n",
              "      <td>WP271</td>\n",
              "      <td>0/0</td>\n",
              "      <td>38.0</td>\n",
              "      <td>38,0</td>\n",
              "      <td>0.031</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>16</th>\n",
              "      <td>chr20</td>\n",
              "      <td>32434599</td>\n",
              "      <td>TCACCACTGCCATAGAGAGGCGGC</td>\n",
              "      <td>T</td>\n",
              "      <td>chr20:32434600-32434622</td>\n",
              "      <td>ASXL1</td>\n",
              "      <td>frameshift_variant</td>\n",
              "      <td>NM_015338.6</td>\n",
              "      <td>ENST00000375687.10</td>\n",
              "      <td>protein_coding</td>\n",
              "      <td>NM_015338.6:c.1900_1922del</td>\n",
              "      <td>NP_056153.2:p.Glu635ArgfsTer15</td>\n",
              "      <td>13/13</td>\n",
              "      <td>.</td>\n",
              "      <td>deletion</td>\n",
              "      <td>.</td>\n",
              "      <td>.</td>\n",
              "      <td>0.000113</td>\n",
              "      <td>0.000236</td>\n",
              "      <td>HIGH</td>\n",
              "      <td>likely_pathogenic&amp;pathogenic</td>\n",
              "      <td>.</td>\n",
              "      <td>rs766433101</td>\n",
              "      <td>PASS</td>\n",
              "      <td>WP270</td>\n",
              "      <td>0/1</td>\n",
              "      <td>105</td>\n",
              "      <td>68,37</td>\n",
              "      <td>0.395</td>\n",
              "      <td>WP271</td>\n",
              "      <td>0/0</td>\n",
              "      <td>73.0</td>\n",
              "      <td>73,0</td>\n",
              "      <td>0.015</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-e2189fab-a617-4610-ae77-3ad761f9c62a')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-e2189fab-a617-4610-ae77-3ad761f9c62a button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-e2189fab-a617-4610-ae77-3ad761f9c62a');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-ab917e0f-2cac-4cac-9645-2f3060c2171a\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-ab917e0f-2cac-4cac-9645-2f3060c2171a')\"\n",
              "            title=\"Suggest charts\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "      --bg-color: #E8F0FE;\n",
              "      --fill-color: #1967D2;\n",
              "      --hover-bg-color: #E2EBFA;\n",
              "      --hover-fill-color: #174EA6;\n",
              "      --disabled-fill-color: #AAA;\n",
              "      --disabled-bg-color: #DDD;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "      --bg-color: #3B4455;\n",
              "      --fill-color: #D2E3FC;\n",
              "      --hover-bg-color: #434B5C;\n",
              "      --hover-fill-color: #FFFFFF;\n",
              "      --disabled-bg-color: #3B4455;\n",
              "      --disabled-fill-color: #666;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart {\n",
              "    background-color: var(--bg-color);\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: var(--fill-color);\n",
              "    height: 32px;\n",
              "    padding: 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: var(--hover-bg-color);\n",
              "    box-shadow: 0 1px 2px rgba(60, 64, 67, 0.3), 0 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: var(--button-hover-fill-color);\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart-complete:disabled,\n",
              "  .colab-df-quickchart-complete:disabled:hover {\n",
              "    background-color: var(--disabled-bg-color);\n",
              "    fill: var(--disabled-fill-color);\n",
              "    box-shadow: none;\n",
              "  }\n",
              "\n",
              "  .colab-df-spinner {\n",
              "    border: 2px solid var(--fill-color);\n",
              "    border-color: transparent;\n",
              "    border-bottom-color: var(--fill-color);\n",
              "    animation:\n",
              "      spin 1s steps(1) infinite;\n",
              "  }\n",
              "\n",
              "  @keyframes spin {\n",
              "    0% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "      border-left-color: var(--fill-color);\n",
              "    }\n",
              "    20% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    30% {\n",
              "      border-color: transparent;\n",
              "      border-left-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    40% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-top-color: var(--fill-color);\n",
              "    }\n",
              "    60% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "    }\n",
              "    80% {\n",
              "      border-color: transparent;\n",
              "      border-right-color: var(--fill-color);\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "    90% {\n",
              "      border-color: transparent;\n",
              "      border-bottom-color: var(--fill-color);\n",
              "    }\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const quickchartButtonEl =\n",
              "        document.querySelector('#' + key + ' button');\n",
              "      quickchartButtonEl.disabled = true;  // To prevent multiple clicks.\n",
              "      quickchartButtonEl.classList.add('colab-df-spinner');\n",
              "      try {\n",
              "        const charts = await google.colab.kernel.invokeFunction(\n",
              "            'suggestCharts', [key], {});\n",
              "      } catch (error) {\n",
              "        console.error('Error during call to suggestCharts:', error);\n",
              "      }\n",
              "      quickchartButtonEl.classList.remove('colab-df-spinner');\n",
              "      quickchartButtonEl.classList.add('colab-df-quickchart-complete');\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-ab917e0f-2cac-4cac-9645-2f3060c2171a button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "\n",
              "  <div id=\"id_d337df1e-86fe-4c8d-97e7-5ba1eea876b0\">\n",
              "    <style>\n",
              "      .colab-df-generate {\n",
              "        background-color: #E8F0FE;\n",
              "        border: none;\n",
              "        border-radius: 50%;\n",
              "        cursor: pointer;\n",
              "        display: none;\n",
              "        fill: #1967D2;\n",
              "        height: 32px;\n",
              "        padding: 0 0 0 0;\n",
              "        width: 32px;\n",
              "      }\n",
              "\n",
              "      .colab-df-generate:hover {\n",
              "        background-color: #E2EBFA;\n",
              "        box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "        fill: #174EA6;\n",
              "      }\n",
              "\n",
              "      [theme=dark] .colab-df-generate {\n",
              "        background-color: #3B4455;\n",
              "        fill: #D2E3FC;\n",
              "      }\n",
              "\n",
              "      [theme=dark] .colab-df-generate:hover {\n",
              "        background-color: #434B5C;\n",
              "        box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "        filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "        fill: #FFFFFF;\n",
              "      }\n",
              "    </style>\n",
              "    <button class=\"colab-df-generate\" onclick=\"generateWithVariable('df_concat')\"\n",
              "            title=\"Generate code using this dataframe.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M7,19H8.4L18.45,9,17,7.55,7,17.6ZM5,21V16.75L18.45,3.32a2,2,0,0,1,2.83,0l1.4,1.43a1.91,1.91,0,0,1,.58,1.4,1.91,1.91,0,0,1-.58,1.4L9.25,21ZM18.45,9,17,7.55Zm-12,3A5.31,5.31,0,0,0,4.9,8.1,5.31,5.31,0,0,0,1,6.5,5.31,5.31,0,0,0,4.9,4.9,5.31,5.31,0,0,0,6.5,1,5.31,5.31,0,0,0,8.1,4.9,5.31,5.31,0,0,0,12,6.5,5.46,5.46,0,0,0,6.5,12Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "    <script>\n",
              "      (() => {\n",
              "      const buttonEl =\n",
              "        document.querySelector('#id_d337df1e-86fe-4c8d-97e7-5ba1eea876b0 button.colab-df-generate');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      buttonEl.onclick = () => {\n",
              "        google.colab.notebook.generateWithVariable('df_concat');\n",
              "      }\n",
              "      })();\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "    </div>\n",
              "  </div>\n"
            ]
          },
          "metadata": {},
          "execution_count": 12
        }
      ],
      "source": [
        "import glob\n",
        "import pandas as pd\n",
        "\n",
        "# Lista pasta com os 30 arquivos csv(s)\n",
        "csv_files = glob.glob('/content/lmabrasil-hg38/csv_filtrados/*.{}'.format('csv'))\n",
        "\n",
        "#Une o resultado dos 30 arquivos csv(s) numa única tabela\n",
        "df_concat = pd.concat([pd.read_csv(i) for i in csv_files], ignore_index=True)\n",
        "\n",
        "#OUTPUT TABELA FINAL EM CSV\n",
        "!mkdir /content/lmabrasil-hg38/tabela_final\n",
        "df_concat.to_csv('/content/lmabrasil-hg38/tabela_final/tabela_final.csv', index=False)\n",
        "\n",
        "df_concat"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "TmtvG8I6nEVK"
      },
      "source": [
        "## **7. TABELA COM SCORES DE RISCO - EM CRIAÇÃO**\n"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "1hzTmhJXm9TG"
      },
      "source": []
    },
    {
      "cell_type": "code",
      "source": [
        "import pandas as pd\n",
        "pd.set_option('display.max_columns', None)\n",
        "pd.set_option('display.max_colwidth', None)\n",
        "\n",
        "df = pd.read_csv(\"/content/lmabrasil-hg38/tabela_final/tabela_final.csv\")\n",
        "#Extrai da tabela_final, em um csv resumido contendo informações das colunas \"TumorID\",\"HGVSc\",\"SYMBOL\", \"HGVSp\", \"CHROM\", \"EXON\"\n",
        "df1 = df[[\"TumorID\",\"SYMBOL\", \"HGVSc\",\"HGVSp\", \"CHROM\", \"EXON\"]]\n",
        "df1.to_csv('/content/tabela_resumo_.csv', index=False)\n",
        "\n",
        "#Converte a tabela resumida em uma lista\n",
        "lista = df1.values.tolist()\n",
        "S1 = []\n",
        "variantes = []\n",
        "var = []\n",
        "\n",
        "#SEPARANDO O NOME DAS AMOSTRAS NÃO ENCONTRADAS NO FILTRO DE VARIANTES, NA VARIÁVEL S1, QUE COMPUTARÁ TODOS COMO +1 DEVIDO A AUSÊNCIA DE CALR TIPO1 PROTETOR)\n",
        "for x in range(len(lista)):\n",
        "  var = lista[x][0]\n",
        "  variantes.append(var)\n",
        "\n",
        "for i in SAMPLES:\n",
        "  if i not in variantes:\n",
        "    S1.append(i)\n",
        "\n",
        "#dicionário para receber variantes, todos começam com pontuação +1, só são despontuados em -1 se for encontrado mutação tipo 1 em CALR\n",
        "#PONTUA TODAS AS AMOSTRAS DA LISTA DE VARIANTES CONFORME CRITÉRIOS DA GISSP + GENES IDH1, IDH2 E EZH2\n",
        "dicio = {}\n",
        "for i in range(len(lista)):\n",
        "  dicio[lista[i][0]] = 1\n",
        "\n",
        "\n",
        "for i in range(len(lista)):\n",
        "  if (\"CALR\" and \"NM_004343.4:c.1099_1150del\") in lista[i]: #mutação do tipo 1 em CARL\n",
        "    for k, v in dicio.items():\n",
        "      if lista[i][0] == k:\n",
        "        dicio[lista[i][0]] +=-1\n",
        "\n",
        "  if (\"SRSF2\" and \"1/3\") in lista[i]: #hostspot em exon 1 de SRSF2\n",
        "    for k, v in dicio.items():\n",
        "      if lista[i][0] == k:\n",
        "        dicio[lista[i][0]] +=1\n",
        "\n",
        "  if \"ASXL1\" in lista[i]: #hotspot cromossomo 12?\n",
        "    for k, v in dicio.items():\n",
        "      if lista[i][0] == k:\n",
        "        dicio[lista[i][0]] +=1\n",
        "\n",
        "  if (\"U2AF1\" and \"NM_006758.3:c.470A>C\") in lista[i]: #hotspot mutação Q157 EM U2AF1\n",
        "    for k, v in dicio.items():\n",
        "      if lista[i][0] == k:\n",
        "        dicio[lista[i][0]] +=1\n",
        "\n",
        "for i in range(len(S1)):\n",
        "  dicio[S1[i]] = 1\n",
        "\n",
        "\n",
        "#Computando as pontuações usando um dicionário para armazenar as contas\n",
        "\n",
        "contagem = {}\n",
        "\n",
        "for i in dicio.values():\n",
        "  if i in contagem:\n",
        "    contagem[i] +=1\n",
        "  else:\n",
        "    contagem[i] =1\n",
        "\n",
        "#Exibindo os dados usando um dataframe pandas\n",
        "\n",
        "import pandas as pd\n",
        "\n",
        "data = {\"TOTAL DE PACIENTES\": [contagem[0],contagem[1],contagem[2],contagem[3]],\n",
        "        \"SCORE\": [\"0\", \"1\", \"2\", \">=3\"],\n",
        "        \"RISCO\": [\"Baixo\", \"Intermediário 1\", \"Intermediário 2\", \"Alto\"],\n",
        "        \"SOBREVIDA EM 5 ANOS (%)\": [\"94,0%\", \"73,0%\", \"40,0%\", \"14,0%\"],\n",
        "        \"SOBREVIDA MÉDIA (ANOS)\": [\"26,4 anos\", \"10,3 anos\", \"4,6 anos\", \"2,6 anos\"]}\n",
        "\n",
        "dataf = pd.DataFrame(data)\n",
        "\n",
        "dataf.style.set_caption('ESTRATIFICADOR DE PROGNÓSTICO DE MIELOFIBROSE BASEADADO EM GIPSS (sem cariotipagem)')"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 192
        },
        "id": "o-FrluL58JXq",
        "outputId": "c630bf39-0930-4009-d57b-0a22601158b7"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "<pandas.io.formats.style.Styler at 0x7f1868fb81c0>"
            ],
            "text/html": [
              "<style type=\"text/css\">\n",
              "</style>\n",
              "<table id=\"T_047e9\" class=\"dataframe\">\n",
              "  <caption>ESTRATIFICADOR DE PROGNÓSTICO DE MIELOFIBROSE BASEADADO EM GIPSS (sem cariotipagem)</caption>\n",
              "  <thead>\n",
              "    <tr>\n",
              "      <th class=\"blank level0\" >&nbsp;</th>\n",
              "      <th id=\"T_047e9_level0_col0\" class=\"col_heading level0 col0\" >PACIENTES</th>\n",
              "      <th id=\"T_047e9_level0_col1\" class=\"col_heading level0 col1\" >SCORE</th>\n",
              "      <th id=\"T_047e9_level0_col2\" class=\"col_heading level0 col2\" >RISCO</th>\n",
              "      <th id=\"T_047e9_level0_col3\" class=\"col_heading level0 col3\" >SOBREVIDA EM 5 ANOS (%)</th>\n",
              "      <th id=\"T_047e9_level0_col4\" class=\"col_heading level0 col4\" >SOBREVIDA MÉDIA (ANOS)</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th id=\"T_047e9_level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
              "      <td id=\"T_047e9_row0_col0\" class=\"data row0 col0\" >4</td>\n",
              "      <td id=\"T_047e9_row0_col1\" class=\"data row0 col1\" >0</td>\n",
              "      <td id=\"T_047e9_row0_col2\" class=\"data row0 col2\" >Baixo</td>\n",
              "      <td id=\"T_047e9_row0_col3\" class=\"data row0 col3\" >94,0%</td>\n",
              "      <td id=\"T_047e9_row0_col4\" class=\"data row0 col4\" >26,4 anos</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th id=\"T_047e9_level0_row1\" class=\"row_heading level0 row1\" >1</th>\n",
              "      <td id=\"T_047e9_row1_col0\" class=\"data row1 col0\" >21</td>\n",
              "      <td id=\"T_047e9_row1_col1\" class=\"data row1 col1\" >1</td>\n",
              "      <td id=\"T_047e9_row1_col2\" class=\"data row1 col2\" >Intermediário 1</td>\n",
              "      <td id=\"T_047e9_row1_col3\" class=\"data row1 col3\" >73,0%</td>\n",
              "      <td id=\"T_047e9_row1_col4\" class=\"data row1 col4\" >10,3 anos</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th id=\"T_047e9_level0_row2\" class=\"row_heading level0 row2\" >2</th>\n",
              "      <td id=\"T_047e9_row2_col0\" class=\"data row2 col0\" >4</td>\n",
              "      <td id=\"T_047e9_row2_col1\" class=\"data row2 col1\" >2</td>\n",
              "      <td id=\"T_047e9_row2_col2\" class=\"data row2 col2\" >Intermediário 2</td>\n",
              "      <td id=\"T_047e9_row2_col3\" class=\"data row2 col3\" >40,0%</td>\n",
              "      <td id=\"T_047e9_row2_col4\" class=\"data row2 col4\" >4,6 anos</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th id=\"T_047e9_level0_row3\" class=\"row_heading level0 row3\" >3</th>\n",
              "      <td id=\"T_047e9_row3_col0\" class=\"data row3 col0\" >1</td>\n",
              "      <td id=\"T_047e9_row3_col1\" class=\"data row3 col1\" >>=3</td>\n",
              "      <td id=\"T_047e9_row3_col2\" class=\"data row3 col2\" >Alto</td>\n",
              "      <td id=\"T_047e9_row3_col3\" class=\"data row3 col3\" >14,0%</td>\n",
              "      <td id=\"T_047e9_row3_col4\" class=\"data row3 col4\" >2,6 anos</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n"
            ]
          },
          "metadata": {},
          "execution_count": 14
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "\n",
        "criterios = {\"Penalidade\": [\"+1\", \"+1\", \"+1\", \"+1\"],\n",
        "        \"Achados\": [\"Ausência de deleção de 52pb em CALR\", \"Mutação em Exon 1 de SRSF2\", \"Mutação em ASXL1\", \"Mutação Q157 em U2AF1\"]}\n",
        "criteriosf = pd.DataFrame(criterios)\n",
        "criteriosf.style.set_caption('CRITÉRIOS DE PONTUAÇÃO BASEADO EM GIPSS (adaptado)')"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 210
        },
        "id": "etieYQkDCIc6",
        "outputId": "b6f2d131-590a-4614-9f6e-1a8c774aabf4"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "<pandas.io.formats.style.Styler at 0x7f1868fe8820>"
            ],
            "text/html": [
              "<style type=\"text/css\">\n",
              "</style>\n",
              "<table id=\"T_e01bb\" class=\"dataframe\">\n",
              "  <caption>CRITÉRIOS DE PONTUAÇÃO BASEADO EM GIPSS (adaptado)</caption>\n",
              "  <thead>\n",
              "    <tr>\n",
              "      <th class=\"blank level0\" >&nbsp;</th>\n",
              "      <th id=\"T_e01bb_level0_col0\" class=\"col_heading level0 col0\" >Penalidade</th>\n",
              "      <th id=\"T_e01bb_level0_col1\" class=\"col_heading level0 col1\" >Achados</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th id=\"T_e01bb_level0_row0\" class=\"row_heading level0 row0\" >0</th>\n",
              "      <td id=\"T_e01bb_row0_col0\" class=\"data row0 col0\" >+1</td>\n",
              "      <td id=\"T_e01bb_row0_col1\" class=\"data row0 col1\" >Ausência de deleção de 52pb em CALR</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th id=\"T_e01bb_level0_row1\" class=\"row_heading level0 row1\" >1</th>\n",
              "      <td id=\"T_e01bb_row1_col0\" class=\"data row1 col0\" >+1</td>\n",
              "      <td id=\"T_e01bb_row1_col1\" class=\"data row1 col1\" >Mutação em Exon 1 de SRSF2</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th id=\"T_e01bb_level0_row2\" class=\"row_heading level0 row2\" >2</th>\n",
              "      <td id=\"T_e01bb_row2_col0\" class=\"data row2 col0\" >+1</td>\n",
              "      <td id=\"T_e01bb_row2_col1\" class=\"data row2 col1\" >Mutação em ASXL1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th id=\"T_e01bb_level0_row3\" class=\"row_heading level0 row3\" >3</th>\n",
              "      <td id=\"T_e01bb_row3_col0\" class=\"data row3 col0\" >+1</td>\n",
              "      <td id=\"T_e01bb_row3_col1\" class=\"data row3 col1\" >Mutação Q157 em U2AF1</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n"
            ]
          },
          "metadata": {},
          "execution_count": 15
        }
      ]
    }
  ],
  "metadata": {
    "colab": {
      "provenance": [],
      "include_colab_link": true
    },
    "kernelspec": {
      "display_name": "Python 3",
      "name": "python3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 0
}